// angular-epic-spinners/esm2015/src/app/fulfilling-square-spinner/fulfilling-square-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class FulfillingSquareSpinnerComponent {
    constructor() {
        this.animationDuration = 4000;
        this.size = 50;
        this.color = '#fff';
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`,
            borderColor: this.color,
        };
    }
    /**
     * @return {?}
     */
    get spinnerInnerStyle() {
        return {
            backgroundColor: this.color
        };
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
FulfillingSquareSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fulfilling-square-spinner',
                template: `
    <div class="fulfilling-square-spinner" [ngStyle]="spinnerStyle">
      <div class="spinner-inner" [ngStyle]="spinnerInnerStyle"></div>
    </div>
  `,
                styles: [".fulfilling-square-spinner,.fulfilling-square-spinner *{box-sizing:border-box}.fulfilling-square-spinner{height:50px;width:50px;position:relative;border:4px solid #ff1d5e;-webkit-animation:4s infinite fulfilling-square-spinner-animation;animation:4s infinite fulfilling-square-spinner-animation}.fulfilling-square-spinner .spinner-inner{vertical-align:top;display:inline-block;background-color:#ff1d5e;width:100%;opacity:1;-webkit-animation:4s ease-in infinite fulfilling-square-spinner-inner-animation;animation:4s ease-in infinite fulfilling-square-spinner-inner-animation}@-webkit-keyframes fulfilling-square-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}25%,50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}100%,75%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes fulfilling-square-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}25%,50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}100%,75%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@-webkit-keyframes fulfilling-square-spinner-inner-animation{0%,100%,25%{height:0}50%,75%{height:100%}}@keyframes fulfilling-square-spinner-inner-animation{0%,100%,25%{height:0}50%,75%{height:100%}}"]
            }] }
];
/** @nocollapse */
FulfillingSquareSpinnerComponent.ctorParameters = () => [];
FulfillingSquareSpinnerComponent.propDecorators = {
    animationDuration: [{ type: Input }],
    size: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    FulfillingSquareSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    FulfillingSquareSpinnerComponent.prototype.size;
    /** @type {?} */
    FulfillingSquareSpinnerComponent.prototype.color;
}
