// angular-epic-spinners/bundles/angular-epic-spinners.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('angular-epic-spinners', ['exports', '@angular/core', '@angular/common'], factory) :
    (factory((global['angular-epic-spinners'] = {}),global.ng.core,global.ng.common));
}(this, (function (exports,core,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var AtomSpinnerComponent = /** @class */ (function () {
        function AtomSpinnerComponent() {
            this.size = 60;
            this.animationDuration = 1000;
            this.color = 'red';
        }
        Object.defineProperty(AtomSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(AtomSpinnerComponent.prototype, "circleStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    color: this.color,
                    fontSize: this.size * 0.24 + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(AtomSpinnerComponent.prototype, "lineStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    animationDuration: this.animationDuration + "ms",
                    borderLeftWidth: this.size / 25 + "px",
                    borderRightWidth: this.size / 25 + "px",
                    borderLeftColor: this.color
                };
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        AtomSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        AtomSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-atom-spinner',
                        template: "\n      <div class=\"atom-spinner\" [ngStyle]=\"spinnerStyle\">\n        <div class=\"spinner-inner\">\n          <div class=\"spinner-line\" [ngStyle]=\"lineStyle\"></div>\n          <div class=\"spinner-line\" [ngStyle]=\"lineStyle\"></div>\n          <div class=\"spinner-line\" [ngStyle]=\"lineStyle\"></div>\n          <div class=\"spinner-circle\" [ngStyle]=\"circleStyle\">\n            &#9679;\n          </div>\n        </div>\n      </div>\n  ",
                        styles: [".atom-spinner,.atom-spinner *{box-sizing:border-box}.atom-spinner{height:60px;width:60px;overflow:hidden}.atom-spinner .spinner-inner{position:relative;display:block;height:100%;width:100%}.atom-spinner .spinner-circle{display:block;position:absolute;color:#ff1d5e;font-size:calc(60px * .24);top:50%;left:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%)}.atom-spinner .spinner-line{position:absolute;width:100%;height:100%;border-radius:50%;-webkit-animation-duration:1s;animation-duration:1s;border-left-width:calc(60px / 25);border-top-width:calc(60px / 25);border-left-color:#ff1d5e;border-left-style:solid;border-top-style:solid;border-top-color:transparent}.atom-spinner .spinner-line:nth-child(1){-webkit-animation:1s linear infinite atom-spinner-animation-1;animation:1s linear infinite atom-spinner-animation-1;-webkit-transform:rotateZ(120deg) rotateX(66deg) rotateZ(0);transform:rotateZ(120deg) rotateX(66deg) rotateZ(0)}.atom-spinner .spinner-line:nth-child(2){-webkit-animation:1s linear infinite atom-spinner-animation-2;animation:1s linear infinite atom-spinner-animation-2;-webkit-transform:rotateZ(240deg) rotateX(66deg) rotateZ(0);transform:rotateZ(240deg) rotateX(66deg) rotateZ(0)}.atom-spinner .spinner-line:nth-child(3){-webkit-animation:1s linear infinite atom-spinner-animation-3;animation:1s linear infinite atom-spinner-animation-3;-webkit-transform:rotateZ(360deg) rotateX(66deg) rotateZ(0);transform:rotateZ(360deg) rotateX(66deg) rotateZ(0)}@-webkit-keyframes atom-spinner-animation-1{100%{-webkit-transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg)}}@keyframes atom-spinner-animation-1{100%{-webkit-transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg)}}@-webkit-keyframes atom-spinner-animation-2{100%{-webkit-transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg)}}@keyframes atom-spinner-animation-2{100%{-webkit-transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg)}}@-webkit-keyframes atom-spinner-animation-3{100%{-webkit-transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg)}}@keyframes atom-spinner-animation-3{100%{-webkit-transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        AtomSpinnerComponent.ctorParameters = function () { return []; };
        AtomSpinnerComponent.propDecorators = {
            size: [{ type: core.Input }],
            animationDuration: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return AtomSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var AtomSpinnerModule = /** @class */ (function () {
        function AtomSpinnerModule() {
        }
        AtomSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [AtomSpinnerComponent],
                        declarations: [AtomSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return AtomSpinnerModule;
    }());

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var BreedingRhombusSpinnerComponent = /** @class */ (function () {
        function BreedingRhombusSpinnerComponent() {
            this.size = 150;
            this.animationDuration = 2000;
            this.color = '#fff';
            this.animationBaseName = 'breeding-rhombus-spinner-animation-child';
            this.rhombusesNum = 8;
        }
        Object.defineProperty(BreedingRhombusSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(BreedingRhombusSpinnerComponent.prototype, "rhombusStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size / 7.5 + "px",
                    width: this.size / 7.5 + "px",
                    animationDuration: "" + this.animationDuration,
                    top: this.size / 2.3077 + "px",
                    left: this.size / 23077 + "px",
                    backgroundColor: this.color
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(BreedingRhombusSpinnerComponent.prototype, "rhombusesStyles", {
            get: /**
             * @return {?}
             */ function () {
                /** @type {?} */
                var rhombusesStyles = [];
                /** @type {?} */
                var delayModifier = this.animationDuration * 0.05;
                for (var i = 1; i <= this.rhombusesNum; i++) {
                    rhombusesStyles.push(__assign({}, this.rhombusStyle, { animationDelay: delayModifier * (i + 1) + "ms" }));
                }
                return rhombusesStyles;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(BreedingRhombusSpinnerComponent.prototype, "bigRhombusStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size / 3 + "px",
                    width: this.size / 3 + "px",
                    animationDuration: "" + this.animationDuration,
                    top: this.size / 3 + "px",
                    left: this.size / 3 + "px",
                    backgroundColor: this.color
                };
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @param {?} index
         * @return {?}
         */
        BreedingRhombusSpinnerComponent.prototype.childClassString = /**
         * @param {?} index
         * @return {?}
         */
            function (index) {
                return "child-" + (index + 1);
            };
        /**
         * @return {?}
         */
        BreedingRhombusSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        BreedingRhombusSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-breeding-rhombus-spinner',
                        template: "\n      <div class=\"breeding-rhombus-spinner\" [ngStyle]=\"spinnerStyle\">\n        <div class=\"rhombus\" *ngFor=\"let rs of rhombusesStyles; let i = index;\" [ngStyle]=\"rs\" [ngClass]=\"childClassString(i)\"></div>\n        <div class=\"rhombus big\" [ngStyle]=\"bigRhombusStyle\"></div>\n      </div>\n    ",
                        styles: [".breeding-rhombus-spinner{height:65px;width:65px;position:relative;-webkit-transform:rotate(45deg);transform:rotate(45deg)}.breeding-rhombus-spinner,.breeding-rhombus-spinner *{box-sizing:border-box}.breeding-rhombus-spinner .rhombus{height:calc(65px / 7.5);width:calc(65px / 7.5);-webkit-animation-duration:2s;animation-duration:2s;top:calc(65px / 2.3077);left:calc(65px / 2.3077);background-color:#ff1d5e;position:absolute;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite}.breeding-rhombus-spinner .rhombus:nth-child(2n+0){margin-right:0}.breeding-rhombus-spinner .rhombus.child-1{-webkit-animation-name:breeding-rhombus-spinner-animation-child-1;animation-name:breeding-rhombus-spinner-animation-child-1;-webkit-animation-delay:calc(100ms * 1);animation-delay:calc(100ms * 1)}.breeding-rhombus-spinner .rhombus.child-2{-webkit-animation-name:breeding-rhombus-spinner-animation-child-2;animation-name:breeding-rhombus-spinner-animation-child-2;-webkit-animation-delay:calc(100ms * 2);animation-delay:calc(100ms * 2)}.breeding-rhombus-spinner .rhombus.child-3{-webkit-animation-name:breeding-rhombus-spinner-animation-child-3;animation-name:breeding-rhombus-spinner-animation-child-3;-webkit-animation-delay:calc(100ms * 3);animation-delay:calc(100ms * 3)}.breeding-rhombus-spinner .rhombus.child-4{-webkit-animation-name:breeding-rhombus-spinner-animation-child-4;animation-name:breeding-rhombus-spinner-animation-child-4;-webkit-animation-delay:calc(100ms * 4);animation-delay:calc(100ms * 4)}.breeding-rhombus-spinner .rhombus.child-5{-webkit-animation-name:breeding-rhombus-spinner-animation-child-5;animation-name:breeding-rhombus-spinner-animation-child-5;-webkit-animation-delay:calc(100ms * 5);animation-delay:calc(100ms * 5)}.breeding-rhombus-spinner .rhombus.child-6{-webkit-animation-name:breeding-rhombus-spinner-animation-child-6;animation-name:breeding-rhombus-spinner-animation-child-6;-webkit-animation-delay:calc(100ms * 6);animation-delay:calc(100ms * 6)}.breeding-rhombus-spinner .rhombus.child-7{-webkit-animation-name:breeding-rhombus-spinner-animation-child-7;animation-name:breeding-rhombus-spinner-animation-child-7;-webkit-animation-delay:calc(100ms * 7);animation-delay:calc(100ms * 7)}.breeding-rhombus-spinner .rhombus.child-8{-webkit-animation-name:breeding-rhombus-spinner-animation-child-8;animation-name:breeding-rhombus-spinner-animation-child-8;-webkit-animation-delay:calc(100ms * 8);animation-delay:calc(100ms * 8)}.breeding-rhombus-spinner .rhombus.big{height:calc(65px / 3);width:calc(65px / 3);top:calc(65px / 3);left:calc(65px / 3);background-color:#ff1d5e;-webkit-animation:2s .5s infinite breeding-rhombus-spinner-animation-child-big;animation:2s .5s infinite breeding-rhombus-spinner-animation-child-big}@-webkit-keyframes breeding-rhombus-spinner-animation-child-1{50%{-webkit-transform:translate(-325%,-325%);transform:translate(-325%,-325%)}}@keyframes breeding-rhombus-spinner-animation-child-1{50%{-webkit-transform:translate(-325%,-325%);transform:translate(-325%,-325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-2{50%{-webkit-transform:translate(0,-325%);transform:translate(0,-325%)}}@keyframes breeding-rhombus-spinner-animation-child-2{50%{-webkit-transform:translate(0,-325%);transform:translate(0,-325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-3{50%{-webkit-transform:translate(325%,-325%);transform:translate(325%,-325%)}}@keyframes breeding-rhombus-spinner-animation-child-3{50%{-webkit-transform:translate(325%,-325%);transform:translate(325%,-325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-4{50%{-webkit-transform:translate(325%,0);transform:translate(325%,0)}}@keyframes breeding-rhombus-spinner-animation-child-4{50%{-webkit-transform:translate(325%,0);transform:translate(325%,0)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-5{50%{-webkit-transform:translate(325%,325%);transform:translate(325%,325%)}}@keyframes breeding-rhombus-spinner-animation-child-5{50%{-webkit-transform:translate(325%,325%);transform:translate(325%,325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-6{50%{-webkit-transform:translate(0,325%);transform:translate(0,325%)}}@keyframes breeding-rhombus-spinner-animation-child-6{50%{-webkit-transform:translate(0,325%);transform:translate(0,325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-7{50%{-webkit-transform:translate(-325%,325%);transform:translate(-325%,325%)}}@keyframes breeding-rhombus-spinner-animation-child-7{50%{-webkit-transform:translate(-325%,325%);transform:translate(-325%,325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-8{50%{-webkit-transform:translate(-325%,0);transform:translate(-325%,0)}}@keyframes breeding-rhombus-spinner-animation-child-8{50%{-webkit-transform:translate(-325%,0);transform:translate(-325%,0)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-big{50%{-webkit-transform:scale(.5);transform:scale(.5)}}@keyframes breeding-rhombus-spinner-animation-child-big{50%{-webkit-transform:scale(.5);transform:scale(.5)}}"]
                    }] }
        ];
        /** @nocollapse */
        BreedingRhombusSpinnerComponent.ctorParameters = function () { return []; };
        BreedingRhombusSpinnerComponent.propDecorators = {
            size: [{ type: core.Input }],
            animationDuration: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return BreedingRhombusSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var BreedingRhombusSpinnerModule = /** @class */ (function () {
        function BreedingRhombusSpinnerModule() {
        }
        BreedingRhombusSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [BreedingRhombusSpinnerComponent],
                        declarations: [BreedingRhombusSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return BreedingRhombusSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var CirclesToRhumbusesSpinnerComponent = /** @class */ (function () {
        function CirclesToRhumbusesSpinnerComponent() {
            this.circleSize = 15;
            this.circlesNum = 3;
            this.animationDuration = 1200;
            this.color = '#fff';
        }
        Object.defineProperty(CirclesToRhumbusesSpinnerComponent.prototype, "circleMarginLeft", {
            get: /**
             * @return {?}
             */ function () {
                return this.circleSize * 1.125;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CirclesToRhumbusesSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.circleSize + "px",
                    width: (this.circleSize + this.circleMarginLeft) * this.circlesNum + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CirclesToRhumbusesSpinnerComponent.prototype, "circleStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    borderColor: this.color,
                    animationDuration: this.animationDuration + "ms",
                    height: this.circleSize + "px",
                    width: this.circleSize + "px",
                    marginLeft: this.circleMarginLeft + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CirclesToRhumbusesSpinnerComponent.prototype, "circlesStyles", {
            get: /**
             * @return {?}
             */ function () {
                /** @type {?} */
                var circlesStyles = [];
                /** @type {?} */
                var delay = 150;
                for (var i = 1; i <= this.circlesNum; i++) {
                    /** @type {?} */
                    var style = __assign({}, this.circleStyle, { animationDelay: i * delay + "ms" });
                    circlesStyles.push(style);
                }
                return circlesStyles;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        CirclesToRhumbusesSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        CirclesToRhumbusesSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-circles-to-rhumbuses-spinner',
                        template: "\n    <div class=\"circles-to-rhombuses-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle\" *ngFor=\"let cs of circlesStyles; let i = index;\" [ngStyle]=\"cs\"></div>\n    </div>\n  ",
                        styles: [".circles-to-rhombuses-spinner,.circles-to-rhombuses-spinner *{box-sizing:border-box}.circles-to-rhombuses-spinner{height:15px;width:calc((15px + 15px * 1.125) * 3);display:flex;align-items:center;justify-content:center}.circles-to-rhombuses-spinner .circle{height:15px;width:15px;margin-left:calc(15px * 1.125);-webkit-transform:rotate(45deg);transform:rotate(45deg);border-radius:10%;border:3px solid #ff1d5e;overflow:hidden;background:0 0;-webkit-animation:1.2s linear infinite circles-to-rhombuses-animation;animation:1.2s linear infinite circles-to-rhombuses-animation}.circles-to-rhombuses-spinner .circle:nth-child(1){-webkit-animation-delay:calc(150ms * 1);animation-delay:calc(150ms * 1);margin-left:0}.circles-to-rhombuses-spinner .circle:nth-child(2){-webkit-animation-delay:calc(150ms * 2);animation-delay:calc(150ms * 2)}.circles-to-rhombuses-spinner .circle:nth-child(3){-webkit-animation-delay:calc(150ms * 3);animation-delay:calc(150ms * 3)}@-webkit-keyframes circles-to-rhombuses-animation{0%,17.5%{border-radius:10%}50%{border-radius:100%}100%,93.5%{border-radius:10%}}@keyframes circles-to-rhombuses-animation{0%,17.5%{border-radius:10%}50%{border-radius:100%}100%,93.5%{border-radius:10%}}@-webkit-keyframes circles-to-rhombuses-background-animation{50%{opacity:.4}}@keyframes circles-to-rhombuses-background-animation{50%{opacity:.4}}"]
                    }] }
        ];
        /** @nocollapse */
        CirclesToRhumbusesSpinnerComponent.ctorParameters = function () { return []; };
        CirclesToRhumbusesSpinnerComponent.propDecorators = {
            circleSize: [{ type: core.Input }],
            circlesNum: [{ type: core.Input }],
            animationDuration: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return CirclesToRhumbusesSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var CirclesToRhumbusesSpinnerModule = /** @class */ (function () {
        function CirclesToRhumbusesSpinnerModule() {
        }
        CirclesToRhumbusesSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [CirclesToRhumbusesSpinnerComponent],
                        declarations: [CirclesToRhumbusesSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return CirclesToRhumbusesSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var FingerprintSpinnerComponent = /** @class */ (function () {
        function FingerprintSpinnerComponent() {
            this.size = 60;
            this.animationDuration = 1500;
            this.color = '#fff';
            this.ringsNum = 9;
            this.containerPadding = 2;
        }
        Object.defineProperty(FingerprintSpinnerComponent.prototype, "outerRingSize", {
            get: /**
             * @return {?}
             */ function () {
                return this.size - this.containerPadding * 2;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(FingerprintSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px",
                    padding: this.containerPadding + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(FingerprintSpinnerComponent.prototype, "ringStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    borderTopColor: this.color,
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(FingerprintSpinnerComponent.prototype, "ringsStyles", {
            get: /**
             * @return {?}
             */ function () {
                /** @type {?} */
                var ringsStyles = [];
                /** @type {?} */
                var ringBase = this.outerRingSize / (this.ringsNum);
                /** @type {?} */
                var ringInc = ringBase;
                for (var i = 1; i <= this.ringsNum; i++) {
                    ringsStyles.push(__assign({}, this.ringStyle, { animationDelay: i * 50 + "ms", height: ringBase + (i + 1) * ringInc + "px", width: ringBase + (i + 1) * ringInc + "px" }));
                }
                return ringsStyles;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        FingerprintSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        FingerprintSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-fingerprint-spinner',
                        template: "\n    <div class=\"fingerprint-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"spinner-ring\" *ngFor=\"let rs of ringsStyles\" [ngStyle]=\"rs\"></div>\n    </div>\n\n  ",
                        styles: [".fingerprint-spinner,.fingerprint-spinner *{box-sizing:border-box}.fingerprint-spinner{height:64px;width:64px;padding:2px;overflow:hidden;position:relative}.fingerprint-spinner .spinner-ring{position:absolute;border-radius:50%;border:2px solid transparent;border-top-color:#ff1d5e;-webkit-animation:1.5s cubic-bezier(.68,-.75,.265,1.75) infinite forwards fingerprint-spinner-animation;animation:1.5s cubic-bezier(.68,-.75,.265,1.75) infinite forwards fingerprint-spinner-animation;margin:auto;bottom:0;left:0;right:0;top:0}.fingerprint-spinner .spinner-ring:nth-child(1){height:calc(60px / 9 + 0 * 60px / 9);width:calc(60px / 9 + 0 * 60px / 9);-webkit-animation-delay:calc(50ms * 1);animation-delay:calc(50ms * 1)}.fingerprint-spinner .spinner-ring:nth-child(2){height:calc(60px / 9 + 1 * 60px / 9);width:calc(60px / 9 + 1 * 60px / 9);-webkit-animation-delay:calc(50ms * 2);animation-delay:calc(50ms * 2)}.fingerprint-spinner .spinner-ring:nth-child(3){height:calc(60px / 9 + 2 * 60px / 9);width:calc(60px / 9 + 2 * 60px / 9);-webkit-animation-delay:calc(50ms * 3);animation-delay:calc(50ms * 3)}.fingerprint-spinner .spinner-ring:nth-child(4){height:calc(60px / 9 + 3 * 60px / 9);width:calc(60px / 9 + 3 * 60px / 9);-webkit-animation-delay:calc(50ms * 4);animation-delay:calc(50ms * 4)}.fingerprint-spinner .spinner-ring:nth-child(5){height:calc(60px / 9 + 4 * 60px / 9);width:calc(60px / 9 + 4 * 60px / 9);-webkit-animation-delay:calc(50ms * 5);animation-delay:calc(50ms * 5)}.fingerprint-spinner .spinner-ring:nth-child(6){height:calc(60px / 9 + 5 * 60px / 9);width:calc(60px / 9 + 5 * 60px / 9);-webkit-animation-delay:calc(50ms * 6);animation-delay:calc(50ms * 6)}.fingerprint-spinner .spinner-ring:nth-child(7){height:calc(60px / 9 + 6 * 60px / 9);width:calc(60px / 9 + 6 * 60px / 9);-webkit-animation-delay:calc(50ms * 7);animation-delay:calc(50ms * 7)}.fingerprint-spinner .spinner-ring:nth-child(8){height:calc(60px / 9 + 7 * 60px / 9);width:calc(60px / 9 + 7 * 60px / 9);-webkit-animation-delay:calc(50ms * 8);animation-delay:calc(50ms * 8)}.fingerprint-spinner .spinner-ring:nth-child(9){height:calc(60px / 9 + 8 * 60px / 9);width:calc(60px / 9 + 8 * 60px / 9);-webkit-animation-delay:calc(50ms * 9);animation-delay:calc(50ms * 9)}@-webkit-keyframes fingerprint-spinner-animation{100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes fingerprint-spinner-animation{100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        FingerprintSpinnerComponent.ctorParameters = function () { return []; };
        FingerprintSpinnerComponent.propDecorators = {
            size: [{ type: core.Input }],
            animationDuration: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return FingerprintSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var FingerprintSpinnerModule = /** @class */ (function () {
        function FingerprintSpinnerModule() {
        }
        FingerprintSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [FingerprintSpinnerComponent],
                        declarations: [FingerprintSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return FingerprintSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var FulfillingBouncingCircleSpinnerComponent = /** @class */ (function () {
        function FulfillingBouncingCircleSpinnerComponent() {
            this.size = 60;
            this.animationDuration = 4000;
            this.color = '#fff';
        }
        Object.defineProperty(FulfillingBouncingCircleSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px",
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(FulfillingBouncingCircleSpinnerComponent.prototype, "orbitStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px",
                    borderColor: this.color,
                    borderWidth: this.size * 0.03 + "px",
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(FulfillingBouncingCircleSpinnerComponent.prototype, "circleStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px",
                    borderColor: this.color,
                    color: this.color,
                    borderWidth: this.size * 0.1 + "px",
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        FulfillingBouncingCircleSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        FulfillingBouncingCircleSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-fulfilling-bouncing-circle-spinner',
                        template: "\n    <div class=\"fulfilling-bouncing-circle-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle\" [ngStyle]=\"circleStyle\"></div>\n      <div class=\"orbit\" [ngStyle]=\"orbitStyle\"></div>\n    </div>\n  ",
                        styles: [".fulfilling-bouncing-circle-spinner,.fulfilling-bouncing-circle-spinner *{box-sizing:border-box}.fulfilling-bouncing-circle-spinner{height:60px;width:60px;position:relative;-webkit-animation:4s infinite fulfilling-bouncing-circle-spinner-animation;animation:4s infinite fulfilling-bouncing-circle-spinner-animation}.fulfilling-bouncing-circle-spinner .orbit{height:60px;width:60px;position:absolute;top:0;left:0;border-radius:50%;border:solid #ff1d5e;-webkit-animation:4s infinite fulfilling-bouncing-circle-spinner-orbit-animation;animation:4s infinite fulfilling-bouncing-circle-spinner-orbit-animation}.fulfilling-bouncing-circle-spinner .circle{height:60px;width:60px;color:#ff1d5e;display:block;border-radius:50%;position:relative;border:solid #ff1d5e;-webkit-animation:4s infinite fulfilling-bouncing-circle-spinner-circle-animation;animation:4s infinite fulfilling-bouncing-circle-spinner-circle-animation;-webkit-transform:rotate(0) scale(1);transform:rotate(0) scale(1)}@-webkit-keyframes fulfilling-bouncing-circle-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes fulfilling-bouncing-circle-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@-webkit-keyframes fulfilling-bouncing-circle-spinner-orbit-animation{0%,50%{-webkit-transform:scale(1);transform:scale(1)}62.5%{-webkit-transform:scale(.8);transform:scale(.8)}75%{-webkit-transform:scale(1);transform:scale(1)}87.5%{-webkit-transform:scale(.8);transform:scale(.8)}100%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes fulfilling-bouncing-circle-spinner-orbit-animation{0%,50%{-webkit-transform:scale(1);transform:scale(1)}62.5%{-webkit-transform:scale(.8);transform:scale(.8)}75%{-webkit-transform:scale(1);transform:scale(1)}87.5%{-webkit-transform:scale(.8);transform:scale(.8)}100%{-webkit-transform:scale(1);transform:scale(1)}}@-webkit-keyframes fulfilling-bouncing-circle-spinner-circle-animation{0%{-webkit-transform:scale(1);transform:scale(1);border-color:transparent;border-top-color:inherit}16.7%{border-color:initial initial transparent transparent}33.4%{border-color:transparent;border-top-color:inherit;border-right-color:inherit;border-bottom-color:inherit}50%{border-color:inherit;-webkit-transform:scale(1);transform:scale(1)}62.5%{border-color:inherit;-webkit-transform:scale(1.4);transform:scale(1.4)}75%{border-color:inherit;-webkit-transform:scale(1);transform:scale(1);opacity:1}87.5%{border-color:inherit;-webkit-transform:scale(1.4);transform:scale(1.4)}100%{border-color:transparent;border-top-color:inherit;-webkit-transform:scale(1);transform:scale(1)}}@keyframes fulfilling-bouncing-circle-spinner-circle-animation{0%{-webkit-transform:scale(1);transform:scale(1);border-color:transparent;border-top-color:inherit}16.7%{border-color:initial initial transparent transparent}33.4%{border-color:transparent;border-top-color:inherit;border-right-color:inherit;border-bottom-color:inherit}50%{border-color:inherit;-webkit-transform:scale(1);transform:scale(1)}62.5%{border-color:inherit;-webkit-transform:scale(1.4);transform:scale(1.4)}75%{border-color:inherit;-webkit-transform:scale(1);transform:scale(1);opacity:1}87.5%{border-color:inherit;-webkit-transform:scale(1.4);transform:scale(1.4)}100%{border-color:transparent;border-top-color:inherit;-webkit-transform:scale(1);transform:scale(1)}}"]
                    }] }
        ];
        /** @nocollapse */
        FulfillingBouncingCircleSpinnerComponent.ctorParameters = function () { return []; };
        FulfillingBouncingCircleSpinnerComponent.propDecorators = {
            size: [{ type: core.Input }],
            animationDuration: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return FulfillingBouncingCircleSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var FulfillingBouncingCircleSpinnerModule = /** @class */ (function () {
        function FulfillingBouncingCircleSpinnerModule() {
        }
        FulfillingBouncingCircleSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [FulfillingBouncingCircleSpinnerComponent],
                        declarations: [FulfillingBouncingCircleSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return FulfillingBouncingCircleSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var FlowerSpinnerComponent = /** @class */ (function () {
        function FlowerSpinnerComponent() {
            this.size = 70;
            this.animationDuration = 2500;
            this.color = '#fff';
            this.smallerDotAnimationBaseName = 'flower-spinner-smaller-dot-animation';
            this.biggerDotAnimationBaseName = 'flower-spinner-bigger-dot-animation';
            this.currentSmallerDotAnimationBaseName = '';
            this.currentBiggerDotAnimationBaseName = '';
        }
        Object.defineProperty(FlowerSpinnerComponent.prototype, "dotSize", {
            get: /**
             * @return {?}
             */ function () {
                return this.size / 7;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(FlowerSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px",
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(FlowerSpinnerComponent.prototype, "dotsContainerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    width: this.color + "px",
                    height: this.color + "px",
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(FlowerSpinnerComponent.prototype, "smallerDotStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    background: this.color,
                    animationDuration: this.animationDuration + "ms",
                    animationName: this.currentSmallerDotAnimationBaseName
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(FlowerSpinnerComponent.prototype, "biggerDotStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    background: this.color,
                    animationDuration: this.animationDuration + "ms",
                    animationName: this.currentBiggerDotAnimationBaseName
                };
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        FlowerSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        /**
         * @return {?}
         */
        FlowerSpinnerComponent.prototype.ngOnChanges = /**
         * @return {?}
         */
            function () {
            };
        FlowerSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-flower-spinner',
                        template: "\n    <div class=\"flower-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"dots-container\" [ngStyle]=\"dotsContainerStyle\">\n        <div class=\"bigger-dot\" [ngStyle]=\"biggerDotStyle\">\n          <div class=\"smaller-dot\" [ngStyle]=\"smallerDotStyle\"></div>\n        </div>\n      </div>\n    </div>\n  ",
                        styles: [".flower-spinner,.flower-spinner *{box-sizing:border-box}.flower-spinner{height:70px;width:70px;display:flex;flex-direction:row;align-items:center;justify-content:center}.flower-spinner .dots-container{height:calc(70px / 7);width:calc(70px / 7)}.flower-spinner .smaller-dot{background:#ff1d5e;height:100%;width:100%;border-radius:50%;-webkit-animation:2.5s infinite both flower-spinner-smaller-dot-animation;animation:2.5s infinite both flower-spinner-smaller-dot-animation}.flower-spinner .bigger-dot{background:#ff1d5e;height:100%;width:100%;padding:10%;border-radius:50%;-webkit-animation:2.5s infinite both flower-spinner-bigger-dot-animation;animation:2.5s infinite both flower-spinner-bigger-dot-animation}@-webkit-keyframes flower-spinner-bigger-dot-animation{0%,100%{box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}25%,75%{box-shadow:#ff1d5e 26px 0 0,#ff1d5e -26px 0 0,#ff1d5e 0 26px 0,#ff1d5e 0 -26px 0,#ff1d5e 19px -19px 0,#ff1d5e 19px 19px 0,#ff1d5e -19px -19px 0,#ff1d5e -19px 19px 0}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg);box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}}@keyframes flower-spinner-bigger-dot-animation{0%,100%{box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}25%,75%{box-shadow:#ff1d5e 26px 0 0,#ff1d5e -26px 0 0,#ff1d5e 0 26px 0,#ff1d5e 0 -26px 0,#ff1d5e 19px -19px 0,#ff1d5e 19px 19px 0,#ff1d5e -19px -19px 0,#ff1d5e -19px 19px 0}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg);box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}}@-webkit-keyframes flower-spinner-smaller-dot-animation{0%,100%{box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}25%,75%{box-shadow:#ff1d5e 14px 0 0,#ff1d5e -14px 0 0,#ff1d5e 0 14px 0,#ff1d5e 0 -14px 0,#ff1d5e 10px -10px 0,#ff1d5e 10px 10px 0,#ff1d5e -10px -10px 0,#ff1d5e -10px 10px 0}}@keyframes flower-spinner-smaller-dot-animation{0%,100%{box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}25%,75%{box-shadow:#ff1d5e 14px 0 0,#ff1d5e -14px 0 0,#ff1d5e 0 14px 0,#ff1d5e 0 -14px 0,#ff1d5e 10px -10px 0,#ff1d5e 10px 10px 0,#ff1d5e -10px -10px 0,#ff1d5e -10px 10px 0}}"]
                    }] }
        ];
        /** @nocollapse */
        FlowerSpinnerComponent.ctorParameters = function () { return []; };
        FlowerSpinnerComponent.propDecorators = {
            size: [{ type: core.Input }],
            animationDuration: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return FlowerSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var FlowerSpinnerModule = /** @class */ (function () {
        function FlowerSpinnerModule() {
        }
        FlowerSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [FlowerSpinnerComponent],
                        declarations: [FlowerSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return FlowerSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var FulfillingSquareSpinnerComponent = /** @class */ (function () {
        function FulfillingSquareSpinnerComponent() {
            this.animationDuration = 4000;
            this.size = 50;
            this.color = '#fff';
        }
        Object.defineProperty(FulfillingSquareSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px",
                    borderColor: this.color,
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(FulfillingSquareSpinnerComponent.prototype, "spinnerInnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    backgroundColor: this.color
                };
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        FulfillingSquareSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        FulfillingSquareSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-fulfilling-square-spinner',
                        template: "\n    <div class=\"fulfilling-square-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"spinner-inner\" [ngStyle]=\"spinnerInnerStyle\"></div>\n    </div>\n  ",
                        styles: [".fulfilling-square-spinner,.fulfilling-square-spinner *{box-sizing:border-box}.fulfilling-square-spinner{height:50px;width:50px;position:relative;border:4px solid #ff1d5e;-webkit-animation:4s infinite fulfilling-square-spinner-animation;animation:4s infinite fulfilling-square-spinner-animation}.fulfilling-square-spinner .spinner-inner{vertical-align:top;display:inline-block;background-color:#ff1d5e;width:100%;opacity:1;-webkit-animation:4s ease-in infinite fulfilling-square-spinner-inner-animation;animation:4s ease-in infinite fulfilling-square-spinner-inner-animation}@-webkit-keyframes fulfilling-square-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}25%,50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}100%,75%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes fulfilling-square-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}25%,50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}100%,75%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@-webkit-keyframes fulfilling-square-spinner-inner-animation{0%,100%,25%{height:0}50%,75%{height:100%}}@keyframes fulfilling-square-spinner-inner-animation{0%,100%,25%{height:0}50%,75%{height:100%}}"]
                    }] }
        ];
        /** @nocollapse */
        FulfillingSquareSpinnerComponent.ctorParameters = function () { return []; };
        FulfillingSquareSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            size: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return FulfillingSquareSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var FulfillingSquareSpinnerModule = /** @class */ (function () {
        function FulfillingSquareSpinnerModule() {
        }
        FulfillingSquareSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [FulfillingSquareSpinnerComponent],
                        declarations: [FulfillingSquareSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return FulfillingSquareSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var HalfCircleSpinnerComponent = /** @class */ (function () {
        function HalfCircleSpinnerComponent() {
            this.animationDuration = 1000;
            this.size = 60;
            this.color = '#fff';
        }
        Object.defineProperty(HalfCircleSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px",
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(HalfCircleSpinnerComponent.prototype, "circleStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    borderWidth: this.size / 10 + "px",
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(HalfCircleSpinnerComponent.prototype, "circle1Style", {
            get: /**
             * @return {?}
             */ function () {
                return __assign({}, this.circleStyle, { borderTopColor: this.color });
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(HalfCircleSpinnerComponent.prototype, "circle2Style", {
            get: /**
             * @return {?}
             */ function () {
                return __assign({}, this.circleStyle, { borderBottomColor: this.color });
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        HalfCircleSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        HalfCircleSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-half-circle-spinner',
                        template: "\n    <div class=\"half-circle-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle circle-1\" [ngStyle]=\"circle1Style\"></div>\n      <div class=\"circle circle-2\" [ngStyle]=\"circle2Style\"></div>\n    </div>\n  ",
                        styles: [".half-circle-spinner,.half-circle-spinner *{box-sizing:border-box}.half-circle-spinner{width:60px;height:60px;border-radius:100%;position:relative}.half-circle-spinner .circle{content:\"\";position:absolute;width:100%;height:100%;border-radius:100%;border:solid transparent}.half-circle-spinner .circle.circle-1{border-top-color:#ff1d5e;-webkit-animation:1s infinite half-circle-spinner-animation;animation:1s infinite half-circle-spinner-animation}.half-circle-spinner .circle.circle-2{border-bottom-color:#ff1d5e;-webkit-animation:1s infinite alternate half-circle-spinner-animation;animation:1s infinite alternate half-circle-spinner-animation}@-webkit-keyframes half-circle-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes half-circle-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        HalfCircleSpinnerComponent.ctorParameters = function () { return []; };
        HalfCircleSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            size: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return HalfCircleSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var HalfCircleSpinnerModule = /** @class */ (function () {
        function HalfCircleSpinnerModule() {
        }
        HalfCircleSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [HalfCircleSpinnerComponent],
                        declarations: [HalfCircleSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return HalfCircleSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var HollowDotsSpinnerComponent = /** @class */ (function () {
        function HollowDotsSpinnerComponent() {
            this.animationDuration = 1000;
            this.dotSize = 15;
            this.dotsNum = 3;
            this.color = '#fff';
        }
        Object.defineProperty(HollowDotsSpinnerComponent.prototype, "horizontalMargin", {
            get: /**
             * @return {?}
             */ function () {
                return this.dotSize / 2;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(HollowDotsSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.dotSize + "px",
                    width: this.dotSize + "px",
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(HollowDotsSpinnerComponent.prototype, "dotStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    width: this.dotSize + "px",
                    height: this.dotSize + "px",
                    margin: "0 " + this.horizontalMargin + "px",
                    borderWidth: this.dotSize / 5 + "px",
                    borderColor: this.color,
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(HollowDotsSpinnerComponent.prototype, "dotsStyles", {
            get: /**
             * @return {?}
             */ function () {
                /** @type {?} */
                var dotsStyles = [];
                /** @type {?} */
                var delayModifier = 0.3;
                /** @type {?} */
                var basicDelay = 1000;
                for (var i = 1; i <= this.dotsNum; i++) {
                    /** @type {?} */
                    var style = __assign({}, this.dotStyle, { animationDelay: basicDelay * i * delayModifier + "ms" });
                    dotsStyles.push(style);
                }
                return dotsStyles;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        HollowDotsSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        HollowDotsSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-hollow-dots-spinner',
                        template: "\n    <div class=\"hollow-dots-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"dot\" *ngFor=\"let ds of dotsStyles\" [ngStyle]=\"ds\"></div>\n    </div>\n  ",
                        styles: [".hollow-dots-spinner,.hollow-dots-spinner *{box-sizing:border-box}.hollow-dots-spinner{height:15px;width:calc(30px * 3)}.hollow-dots-spinner .dot{width:15px;height:15px;margin:0 calc(15px / 2);border:solid #ff1d5e;border-radius:50%;float:left;-webkit-transform:scale(0);transform:scale(0);-webkit-animation:1s infinite hollow-dots-spinner-animation;animation:1s infinite hollow-dots-spinner-animation}.hollow-dots-spinner .dot:nth-child(1){-webkit-animation-delay:calc(300ms * 1);animation-delay:calc(300ms * 1)}.hollow-dots-spinner .dot:nth-child(2){-webkit-animation-delay:calc(300ms * 2);animation-delay:calc(300ms * 2)}.hollow-dots-spinner .dot:nth-child(3){-webkit-animation-delay:calc(300ms * 3);animation-delay:calc(300ms * 3)}@-webkit-keyframes hollow-dots-spinner-animation{50%{-webkit-transform:scale(1);transform:scale(1);opacity:1}100%{opacity:0}}@keyframes hollow-dots-spinner-animation{50%{-webkit-transform:scale(1);transform:scale(1);opacity:1}100%{opacity:0}}"]
                    }] }
        ];
        /** @nocollapse */
        HollowDotsSpinnerComponent.ctorParameters = function () { return []; };
        HollowDotsSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            dotSize: [{ type: core.Input }],
            dotsNum: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return HollowDotsSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var HollowDotsSpinnerModule = /** @class */ (function () {
        function HollowDotsSpinnerModule() {
        }
        HollowDotsSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [HollowDotsSpinnerComponent],
                        declarations: [HollowDotsSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return HollowDotsSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var IntersectingCirclesSpinnerComponent = /** @class */ (function () {
        function IntersectingCirclesSpinnerComponent() {
            this.animationDuration = 1200;
            this.size = 70;
            this.color = '#fff';
        }
        Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "circleSize", {
            get: /**
             * @return {?}
             */ function () {
                return this.size / 2;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    width: this.size + "px",
                    height: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "spinnerBlockStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    animationDuration: this.animationDuration + "ms",
                    width: this.circleSize + "px",
                    height: this.circleSize + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "circleStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    borderColor: this.color
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "circleStyles", {
            get: /**
             * @return {?}
             */ function () {
                var _this = this;
                /** @type {?} */
                var circlesPositions = [
                    { top: 0, left: 0 },
                    { left: this.circleSize * -0.36 + "px", top: this.circleSize * 0.2 + "px" },
                    { left: this.circleSize * -0.36 + "px", top: this.circleSize * -0.2 + "px" },
                    { left: 0, top: this.circleSize * -0.36 + "px" },
                    { left: this.circleSize * 0.36 + "px", top: this.circleSize * -0.2 + "px" },
                    { left: this.circleSize * 0.36 + "px", top: this.circleSize * 0.2 + "px" },
                    { left: 0, top: this.circleSize * 0.36 + "px" }
                ];
                return circlesPositions.map(function (cp) {
                    return __assign({ cp: cp }, _this.circleStyle);
                });
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        IntersectingCirclesSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        IntersectingCirclesSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-intersecting-circles-spinner',
                        template: "\n    <div class=\"intersecting-circles-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"spinnerBlock\" [ngStyle]=\"spinnerBlockStyle\">\n        <span class=\"circle\" *ngFor=\"let cs of circleStyles\" [ngStyle]=\"cs\"></span>\n      </div>\n    </div>\n  ",
                        styles: [".intersecting-circles-spinner,.intersecting-circles-spinner *{box-sizing:border-box}.intersecting-circles-spinner{height:70px;width:70px;position:relative;display:flex;flex-direction:row;justify-content:center;align-items:center}.intersecting-circles-spinner .spinnerBlock{-webkit-animation:1.2s linear infinite intersecting-circles-spinners-animation;animation:1.2s linear infinite intersecting-circles-spinners-animation;-webkit-transform-origin:center;transform-origin:center;display:block;height:35px;width:35px}.intersecting-circles-spinner .circle{display:block;border:2px solid #ff1d5e;border-radius:50%;height:100%;width:100%;position:absolute;left:0;top:0}.intersecting-circles-spinner .circle:nth-child(1){left:0;top:0}.intersecting-circles-spinner .circle:nth-child(2){left:calc(35px * -.36);top:calc(35px * .2)}.intersecting-circles-spinner .circle:nth-child(3){left:calc(35px * -.36);top:calc(35px * -.2)}.intersecting-circles-spinner .circle:nth-child(4){left:0;top:calc(35px * -.36)}.intersecting-circles-spinner .circle:nth-child(5){left:calc(35px * .36);top:calc(35px * -.2)}.intersecting-circles-spinner .circle:nth-child(6){left:calc(35px * .36);top:calc(35px * .2)}.intersecting-circles-spinner .circle:nth-child(7){left:0;top:calc(35px * .36)}@-webkit-keyframes intersecting-circles-spinners-animation{from{-webkit-transform:rotate(0);transform:rotate(0)}to{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes intersecting-circles-spinners-animation{from{-webkit-transform:rotate(0);transform:rotate(0)}to{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        IntersectingCirclesSpinnerComponent.ctorParameters = function () { return []; };
        IntersectingCirclesSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            size: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return IntersectingCirclesSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var IntersectingCirclesSpinnerModule = /** @class */ (function () {
        function IntersectingCirclesSpinnerModule() {
        }
        IntersectingCirclesSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [IntersectingCirclesSpinnerComponent],
                        declarations: [IntersectingCirclesSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return IntersectingCirclesSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var LoopingRhumbusesSpinnerComponent = /** @class */ (function () {
        function LoopingRhumbusesSpinnerComponent() {
            this.animationDuration = 2500;
            this.rhombusSize = 15;
            this.color = '#fff';
            this.rhombusesNum = 3;
        }
        Object.defineProperty(LoopingRhumbusesSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.rhombusSize + "px",
                    width: this.rhombusSize * 4 + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(LoopingRhumbusesSpinnerComponent.prototype, "rhombusStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.rhombusSize + "px",
                    width: this.rhombusSize + "px",
                    background: this.color,
                    animationDuration: this.animationDuration + "ms",
                    left: this.rhombusSize * 4 + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(LoopingRhumbusesSpinnerComponent.prototype, "rhombusesStyles", {
            get: /**
             * @return {?}
             */ function () {
                /** @type {?} */
                var rhombusesStyles = [];
                /** @type {?} */
                var delay = -this.animationDuration / 1.5;
                for (var i = 1; i <= this.rhombusesNum; i++) {
                    /** @type {?} */
                    var style = Object.assign({
                        animationDelay: i * delay + "ms"
                    }, this.rhombusStyle);
                    rhombusesStyles.push(style);
                }
                return rhombusesStyles;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        LoopingRhumbusesSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        LoopingRhumbusesSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-looping-rhombuses-spinner',
                        template: "\n    <div class=\"looping-rhombuses-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"rhombus\"\n           *ngFor=\"let rs of rhombusesStyles\"\n           [ngStyle]=\"rs\">\n      </div>\n    </div>\n  ",
                        styles: [".looping-rhombuses-spinner,.looping-rhombuses-spinner *{box-sizing:border-box}.looping-rhombuses-spinner{width:calc(15px * 4);height:15px;position:relative}.looping-rhombuses-spinner .rhombus{height:15px;width:15px;background:#ff1d5e;left:calc(15px * 4);position:absolute;margin:0 auto;border-radius:2px;-webkit-transform:translateY(0) rotate(45deg) scale(0);transform:translateY(0) rotate(45deg) scale(0);-webkit-animation:2.5s linear infinite looping-rhombuses-spinner-animation;animation:2.5s linear infinite looping-rhombuses-spinner-animation}.looping-rhombuses-spinner .rhombus:nth-child(1){-webkit-animation-delay:calc(2500ms * 1 / -1.5);animation-delay:calc(2500ms * 1 / -1.5)}.looping-rhombuses-spinner .rhombus:nth-child(2){-webkit-animation-delay:calc(2500ms * 2 / -1.5);animation-delay:calc(2500ms * 2 / -1.5)}.looping-rhombuses-spinner .rhombus:nth-child(3){-webkit-animation-delay:calc(2500ms * 3 / -1.5);animation-delay:calc(2500ms * 3 / -1.5)}@-webkit-keyframes looping-rhombuses-spinner-animation{0%{-webkit-transform:translateX(0) rotate(45deg) scale(0);transform:translateX(0) rotate(45deg) scale(0)}50%{-webkit-transform:translateX(-233%) rotate(45deg) scale(1);transform:translateX(-233%) rotate(45deg) scale(1)}100%{-webkit-transform:translateX(-466%) rotate(45deg) scale(0);transform:translateX(-466%) rotate(45deg) scale(0)}}@keyframes looping-rhombuses-spinner-animation{0%{-webkit-transform:translateX(0) rotate(45deg) scale(0);transform:translateX(0) rotate(45deg) scale(0)}50%{-webkit-transform:translateX(-233%) rotate(45deg) scale(1);transform:translateX(-233%) rotate(45deg) scale(1)}100%{-webkit-transform:translateX(-466%) rotate(45deg) scale(0);transform:translateX(-466%) rotate(45deg) scale(0)}}"]
                    }] }
        ];
        /** @nocollapse */
        LoopingRhumbusesSpinnerComponent.ctorParameters = function () { return []; };
        LoopingRhumbusesSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            rhombusSize: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return LoopingRhumbusesSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var LoopingRhumbusesSpinnerModule = /** @class */ (function () {
        function LoopingRhumbusesSpinnerModule() {
        }
        LoopingRhumbusesSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [LoopingRhumbusesSpinnerComponent],
                        declarations: [LoopingRhumbusesSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return LoopingRhumbusesSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var OrbitSpinnerComponent = /** @class */ (function () {
        function OrbitSpinnerComponent() {
            this.animationDuration = 1000;
            this.size = 50;
            this.color = '#fff';
        }
        Object.defineProperty(OrbitSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(OrbitSpinnerComponent.prototype, "orbitStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    borderColor: this.color,
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        OrbitSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        OrbitSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-orbit-spinner',
                        template: "\n    <div class=\"orbit-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"orbit one\" [ngStyle]=\"orbitStyle\"></div>\n      <div class=\"orbit two\" [ngStyle]=\"orbitStyle\"></div>\n      <div class=\"orbit three\" [ngStyle]=\"orbitStyle\"></div>\n    </div>\n  ",
                        styles: [".orbit-spinner,.orbit-spinner *{box-sizing:border-box}.orbit-spinner{height:55px;width:55px;border-radius:50%;-webkit-perspective:800px;perspective:800px}.orbit-spinner .orbit{position:absolute;box-sizing:border-box;width:100%;height:100%;border-radius:50%}.orbit-spinner .orbit:nth-child(1){left:0;top:0;-webkit-animation:1.2s linear infinite orbit-spinner-orbit-one-animation;animation:1.2s linear infinite orbit-spinner-orbit-one-animation;border-bottom:3px solid #ff1d5e}.orbit-spinner .orbit:nth-child(2){right:0;top:0;-webkit-animation:1.2s linear infinite orbit-spinner-orbit-two-animation;animation:1.2s linear infinite orbit-spinner-orbit-two-animation;border-right:3px solid #ff1d5e}.orbit-spinner .orbit:nth-child(3){right:0;bottom:0;-webkit-animation:1.2s linear infinite orbit-spinner-orbit-three-animation;animation:1.2s linear infinite orbit-spinner-orbit-three-animation;border-top:3px solid #ff1d5e}@-webkit-keyframes orbit-spinner-orbit-one-animation{0%{-webkit-transform:rotateX(35deg) rotateY(-45deg) rotateZ(0);transform:rotateX(35deg) rotateY(-45deg) rotateZ(0)}100%{-webkit-transform:rotateX(35deg) rotateY(-45deg) rotateZ(360deg);transform:rotateX(35deg) rotateY(-45deg) rotateZ(360deg)}}@keyframes orbit-spinner-orbit-one-animation{0%{-webkit-transform:rotateX(35deg) rotateY(-45deg) rotateZ(0);transform:rotateX(35deg) rotateY(-45deg) rotateZ(0)}100%{-webkit-transform:rotateX(35deg) rotateY(-45deg) rotateZ(360deg);transform:rotateX(35deg) rotateY(-45deg) rotateZ(360deg)}}@-webkit-keyframes orbit-spinner-orbit-two-animation{0%{-webkit-transform:rotateX(50deg) rotateY(10deg) rotateZ(0);transform:rotateX(50deg) rotateY(10deg) rotateZ(0)}100%{-webkit-transform:rotateX(50deg) rotateY(10deg) rotateZ(360deg);transform:rotateX(50deg) rotateY(10deg) rotateZ(360deg)}}@keyframes orbit-spinner-orbit-two-animation{0%{-webkit-transform:rotateX(50deg) rotateY(10deg) rotateZ(0);transform:rotateX(50deg) rotateY(10deg) rotateZ(0)}100%{-webkit-transform:rotateX(50deg) rotateY(10deg) rotateZ(360deg);transform:rotateX(50deg) rotateY(10deg) rotateZ(360deg)}}@-webkit-keyframes orbit-spinner-orbit-three-animation{0%{-webkit-transform:rotateX(35deg) rotateY(55deg) rotateZ(0);transform:rotateX(35deg) rotateY(55deg) rotateZ(0)}100%{-webkit-transform:rotateX(35deg) rotateY(55deg) rotateZ(360deg);transform:rotateX(35deg) rotateY(55deg) rotateZ(360deg)}}@keyframes orbit-spinner-orbit-three-animation{0%{-webkit-transform:rotateX(35deg) rotateY(55deg) rotateZ(0);transform:rotateX(35deg) rotateY(55deg) rotateZ(0)}100%{-webkit-transform:rotateX(35deg) rotateY(55deg) rotateZ(360deg);transform:rotateX(35deg) rotateY(55deg) rotateZ(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        OrbitSpinnerComponent.ctorParameters = function () { return []; };
        OrbitSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            size: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return OrbitSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var OrbitSpinnerModule = /** @class */ (function () {
        function OrbitSpinnerModule() {
        }
        OrbitSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [OrbitSpinnerComponent],
                        declarations: [OrbitSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return OrbitSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var PixelSpinnerComponent = /** @class */ (function () {
        function PixelSpinnerComponent() {
            this.animationDuration = 1000;
            this.size = 50;
            this.color = '#fff';
            this.animationBaseName = 'pixel-spinner-animation';
            this.currentAnimationName = '';
        }
        Object.defineProperty(PixelSpinnerComponent.prototype, "pixelSize", {
            get: /**
             * @return {?}
             */ function () {
                return this.size / 7;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(PixelSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    width: this.size + "px",
                    height: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(PixelSpinnerComponent.prototype, "spinnerInnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    animationDuration: this.animationDuration + "ms",
                    animationName: this.currentAnimationName,
                    width: this.pixelSize + "px",
                    height: this.pixelSize + "px",
                    backgroundColor: this.color,
                    color: this.color,
                    boxShadow: "\n                        " + this.pixelSize * 1.5 + "px " + this.pixelSize * 1.5 + "px 0 0,\n                        " + this.pixelSize * -1.5 + "px " + this.pixelSize * -1.5 + "px 0 0,\n                        " + this.pixelSize * 1.5 + "px " + this.pixelSize * -1.5 + "px 0 0,\n                        " + this.pixelSize * -1.5 + "px " + this.pixelSize * 1.5 + "px 0 0,\n                        0 " + this.pixelSize * 1.5 + "px 0 0,\n                        " + this.pixelSize * 1.5 + "px 0 0 0,\n                        " + this.pixelSize * -1.5 + "px 0 0 0,\n                        0 " + this.pixelSize * -1.5 + "px 0 0\n                      "
                };
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        PixelSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        PixelSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-pixel-spinner',
                        template: "\n    <div class=\"pixel-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"pixel-spinner-inner\" [ngStyle]=\"spinnerInnerStyle\"></div>\n    </div>\n  ",
                        styles: [".pixel-spinner,.pixel-spinner *{box-sizing:border-box}.pixel-spinner{height:70px;width:70px;display:flex;flex-direction:row;justify-content:center;align-items:center}.pixel-spinner .pixel-spinner-inner{width:calc(70px / 7);height:calc(70px / 7);background-color:#ff1d5e;color:#ff1d5e;box-shadow:15px 15px 0 0,-15px -15px 0 0,15px -15px 0 0,-15px 15px 0 0,0 15px 0 0,15px 0 0 0,-15px 0 0 0,0 -15px 0 0;-webkit-animation:2s linear infinite pixel-spinner-animation;animation:2s linear infinite pixel-spinner-animation}@-webkit-keyframes pixel-spinner-animation{50%,75%{box-shadow:20px 20px 0 0,-20px -20px 0 0,20px -20px 0 0,-20px 20px 0 0,0 10px 0 0,10px 0 0 0,-10px 0 0 0,0 -10px 0 0}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes pixel-spinner-animation{50%,75%{box-shadow:20px 20px 0 0,-20px -20px 0 0,20px -20px 0 0,-20px 20px 0 0,0 10px 0 0,10px 0 0 0,-10px 0 0 0,0 -10px 0 0}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        PixelSpinnerComponent.ctorParameters = function () { return []; };
        PixelSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            size: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return PixelSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var PixelSpinnerModule = /** @class */ (function () {
        function PixelSpinnerModule() {
        }
        PixelSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [PixelSpinnerComponent],
                        declarations: [PixelSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return PixelSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var RadarSpinnerComponent = /** @class */ (function () {
        function RadarSpinnerComponent() {
            this.size = 110;
            this.animationDuration = 2000;
            this.color = '#fff';
            this.circlesNum = 4;
        }
        Object.defineProperty(RadarSpinnerComponent.prototype, "borderWidth", {
            get: /**
             * @return {?}
             */ function () {
                return this.size * 5 / 110;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(RadarSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(RadarSpinnerComponent.prototype, "circleStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(RadarSpinnerComponent.prototype, "circleInnerContainerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    borderWidth: this.borderWidth + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(RadarSpinnerComponent.prototype, "circleInnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    borderLeftColor: this.color,
                    borderRightColor: this.color,
                    borderWidth: this.borderWidth + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(RadarSpinnerComponent.prototype, "circlesStyles", {
            get: /**
             * @return {?}
             */ function () {
                /** @type {?} */
                var circlesStyles = [];
                /** @type {?} */
                var delay = this.animationDuration * 0.15;
                for (var i = 0; i < this.circlesNum; i++) {
                    circlesStyles.push(Object.assign({
                        padding: this.borderWidth * 2 * i + "px",
                        animationDelay: (i === this.circlesNum - 1 ? 0 : delay) + "ms"
                    }, this.circleStyle));
                }
                return circlesStyles;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        RadarSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        RadarSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-radar-spinner',
                        template: "\n    <div class=\"radar-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle\" *ngFor=\"let cs of circlesStyles\" [ngStyle]=\"cs\">\n        <div class=\"circle-inner-container\" [ngStyle]=\"circleInnerContainerStyle\">\n          <div class=\"circle-inner\" [ngStyle]=\"circleInnerStyle\"></div>\n        </div>\n      </div>\n    </div>\n  ",
                        styles: [".radar-spinner,.radar-spinner *{box-sizing:border-box}.radar-spinner{height:60px;width:60px;position:relative}.radar-spinner .circle{position:absolute;height:100%;width:100%;top:0;left:0;-webkit-animation:2s infinite radar-spinner-animation;animation:2s infinite radar-spinner-animation}.radar-spinner .circle:nth-child(1){padding:calc(60px * 5 * 2 * 0 / 110);-webkit-animation-delay:.3s;animation-delay:.3s}.radar-spinner .circle:nth-child(2){padding:calc(60px * 5 * 2 * 1 / 110);-webkit-animation-delay:.3s;animation-delay:.3s}.radar-spinner .circle:nth-child(3){padding:calc(60px * 5 * 2 * 2 / 110);-webkit-animation-delay:.3s;animation-delay:.3s}.radar-spinner .circle:nth-child(4){padding:calc(60px * 5 * 2 * 3 / 110);-webkit-animation-delay:0s;animation-delay:0s}.radar-spinner .circle-inner,.radar-spinner .circle-inner-container{height:100%;width:100%;border-radius:50%;border:solid transparent}.radar-spinner .circle-inner{border-left-color:#ff1d5e;border-right-color:#ff1d5e}@-webkit-keyframes radar-spinner-animation{50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}100%{-webkit-transform:rotate(0);transform:rotate(0)}}@keyframes radar-spinner-animation{50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}100%{-webkit-transform:rotate(0);transform:rotate(0)}}"]
                    }] }
        ];
        /** @nocollapse */
        RadarSpinnerComponent.ctorParameters = function () { return []; };
        RadarSpinnerComponent.propDecorators = {
            size: [{ type: core.Input }],
            animationDuration: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return RadarSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var RadarSpinnerModule = /** @class */ (function () {
        function RadarSpinnerModule() {
        }
        RadarSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [RadarSpinnerComponent],
                        declarations: [RadarSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return RadarSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var ScalingSquaresSpinnerComponent = /** @class */ (function () {
        function ScalingSquaresSpinnerComponent() {
            this.size = 65;
            this.color = '#fff';
            this.animationDuration = 1250;
            this.squaresNum = 4;
        }
        Object.defineProperty(ScalingSquaresSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px",
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ScalingSquaresSpinnerComponent.prototype, "squareStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size * 0.25 / 1.3 + "px",
                    width: this.size * 0.25 / 1.3 + "px",
                    animationDuration: this.animationDuration + "ms",
                    borderWidth: this.size * 0.04 / 1.3 + "px",
                    borderColor: this.color
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ScalingSquaresSpinnerComponent.prototype, "squaresStyles", {
            get: /**
             * @return {?}
             */ function () {
                /** @type {?} */
                var squaresStyles = [];
                for (var i = 1; i <= this.squaresNum; i++) {
                    squaresStyles.push(__assign({}, this.squareStyle));
                }
                return squaresStyles;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @param {?} i
         * @return {?}
         */
        ScalingSquaresSpinnerComponent.prototype.childClass = /**
         * @param {?} i
         * @return {?}
         */
            function (i) {
                return "square-" + (i + 1);
            };
        /**
         * @return {?}
         */
        ScalingSquaresSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        ScalingSquaresSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-scaling-squares-spinner',
                        template: "\n    <div class=\"scaling-squares-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"square\"\n           *ngFor=\"let ss of squaresStyles; let i = index;\"\n           [ngClass]=\"childClass(i)\"\n           [ngStyle]=\"ss\">\n      </div>\n    </div>\n  ",
                        styles: [".scaling-squares-spinner,.scaling-squares-spinner *{box-sizing:border-box}.scaling-squares-spinner{height:65px;width:65px;position:relative;display:flex;flex-direction:row;align-items:center;justify-content:center;-webkit-animation:1.25s infinite scaling-squares-animation;animation:1.25s infinite scaling-squares-animation;-webkit-transform:rotate(0);transform:rotate(0)}.scaling-squares-spinner .square{height:calc(65px * .25 / 1.3);width:calc(65px * .25 / 1.3);margin-right:auto;margin-left:auto;border:solid #ff1d5e;position:absolute;-webkit-animation-duration:1.25s;animation-duration:1.25s;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite}.scaling-squares-spinner .square:nth-child(1){-webkit-animation-name:scaling-squares-spinner-animation-child-1;animation-name:scaling-squares-spinner-animation-child-1}.scaling-squares-spinner .square:nth-child(2){-webkit-animation-name:scaling-squares-spinner-animation-child-2;animation-name:scaling-squares-spinner-animation-child-2}.scaling-squares-spinner .square:nth-child(3){-webkit-animation-name:scaling-squares-spinner-animation-child-3;animation-name:scaling-squares-spinner-animation-child-3}.scaling-squares-spinner .square:nth-child(4){-webkit-animation-name:scaling-squares-spinner-animation-child-4;animation-name:scaling-squares-spinner-animation-child-4}@-webkit-keyframes scaling-squares-animation{50%{-webkit-transform:rotate(90deg);transform:rotate(90deg)}100%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}}@keyframes scaling-squares-animation{50%{-webkit-transform:rotate(90deg);transform:rotate(90deg)}100%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}}@-webkit-keyframes scaling-squares-spinner-animation-child-1{50%{-webkit-transform:translate(150%,150%) scale(2,2);transform:translate(150%,150%) scale(2,2)}}@keyframes scaling-squares-spinner-animation-child-1{50%{-webkit-transform:translate(150%,150%) scale(2,2);transform:translate(150%,150%) scale(2,2)}}@-webkit-keyframes scaling-squares-spinner-animation-child-2{50%{-webkit-transform:translate(-150%,150%) scale(2,2);transform:translate(-150%,150%) scale(2,2)}}@keyframes scaling-squares-spinner-animation-child-2{50%{-webkit-transform:translate(-150%,150%) scale(2,2);transform:translate(-150%,150%) scale(2,2)}}@-webkit-keyframes scaling-squares-spinner-animation-child-3{50%{-webkit-transform:translate(-150%,-150%) scale(2,2);transform:translate(-150%,-150%) scale(2,2)}}@keyframes scaling-squares-spinner-animation-child-3{50%{-webkit-transform:translate(-150%,-150%) scale(2,2);transform:translate(-150%,-150%) scale(2,2)}}@-webkit-keyframes scaling-squares-spinner-animation-child-4{50%{-webkit-transform:translate(150%,-150%) scale(2,2);transform:translate(150%,-150%) scale(2,2)}}@keyframes scaling-squares-spinner-animation-child-4{50%{-webkit-transform:translate(150%,-150%) scale(2,2);transform:translate(150%,-150%) scale(2,2)}}"]
                    }] }
        ];
        /** @nocollapse */
        ScalingSquaresSpinnerComponent.ctorParameters = function () { return []; };
        ScalingSquaresSpinnerComponent.propDecorators = {
            size: [{ type: core.Input }],
            color: [{ type: core.Input }],
            animationDuration: [{ type: core.Input }]
        };
        return ScalingSquaresSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var ScalingSquaresSpinnerModule = /** @class */ (function () {
        function ScalingSquaresSpinnerModule() {
        }
        ScalingSquaresSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [ScalingSquaresSpinnerComponent],
                        declarations: [ScalingSquaresSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return ScalingSquaresSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var SelfBuildingSquareSpinnerComponent = /** @class */ (function () {
        function SelfBuildingSquareSpinnerComponent() {
            this.animationDuration = 6000;
            this.size = 40;
            this.color = '#fff';
            this.squaresNum = 9;
        }
        Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "squareSize", {
            get: /**
             * @return {?}
             */ function () {
                return this.size / 4;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "initialTopPosition", {
            get: /**
             * @return {?}
             */ function () {
                return -this.squareSize * 2 / 3;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    top: -this.initialTopPosition + "px",
                    height: this.size + "px",
                    width: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "squareStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.squareSize + "px",
                    width: this.squareSize + "px",
                    top: this.initialTopPosition + "px",
                    marginRight: this.squareSize / 3 + "px",
                    marginTop: this.squareSize / 3 + "px",
                    animationDuration: this.animationDuration + "ms",
                    background: this.color
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "squaresStyles", {
            get: /**
             * @return {?}
             */ function () {
                /** @type {?} */
                var squaresStyles = [];
                /** @type {?} */
                var delaysMultipliers = [6, 7, 8, 3, 4, 5, 0, 1, 2];
                /** @type {?} */
                var delayModifier = this.animationDuration * 0.05;
                for (var i = 0; i < this.squaresNum; i++) {
                    squaresStyles.push(__assign({ animationDelay: delayModifier * delaysMultipliers[i] + "ms" }, this.squareStyle));
                }
                return squaresStyles;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        SelfBuildingSquareSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        SelfBuildingSquareSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-self-building-square-spinner',
                        template: "\n    <div class=\"self-building-square-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"square\"\n           *ngFor=\"let ss of squaresStyles; let i = index;\"\n           [ngStyle]=\"ss\"\n           [ngClass]=\"{'clear': i && i % 3 === 0}\"\n      >\n      </div>\n    </div>\n  ",
                        styles: [".self-building-square-spinner,.self-building-square-spinner *{box-sizing:border-box}.self-building-square-spinner{height:40px;width:40px;top:calc(-10px * 2 / 3)}.self-building-square-spinner .square{height:10px;width:10px;top:calc(-10px * 2 / 3);margin-right:calc(10px / 3);margin-top:calc(10px / 3);background:#ff1d5e;float:left;position:relative;opacity:0;-webkit-animation:6s infinite self-building-square-spinner;animation:6s infinite self-building-square-spinner}.self-building-square-spinner .square:nth-child(1){-webkit-animation-delay:calc(300ms * 6);animation-delay:calc(300ms * 6)}.self-building-square-spinner .square:nth-child(2){-webkit-animation-delay:calc(300ms * 7);animation-delay:calc(300ms * 7)}.self-building-square-spinner .square:nth-child(3){-webkit-animation-delay:calc(300ms * 8);animation-delay:calc(300ms * 8)}.self-building-square-spinner .square:nth-child(4){-webkit-animation-delay:calc(300ms * 3);animation-delay:calc(300ms * 3)}.self-building-square-spinner .square:nth-child(5){-webkit-animation-delay:calc(300ms * 4);animation-delay:calc(300ms * 4)}.self-building-square-spinner .square:nth-child(6){-webkit-animation-delay:calc(300ms * 5);animation-delay:calc(300ms * 5)}.self-building-square-spinner .square:nth-child(7){-webkit-animation-delay:calc(300ms * 0);animation-delay:calc(300ms * 0)}.self-building-square-spinner .square:nth-child(8){-webkit-animation-delay:calc(300ms * 1);animation-delay:calc(300ms * 1)}.self-building-square-spinner .square:nth-child(9){-webkit-animation-delay:calc(300ms * 2);animation-delay:calc(300ms * 2)}.self-building-square-spinner .clear{clear:both}@-webkit-keyframes self-building-square-spinner{0%{opacity:0}5%,50.9%{opacity:1;top:0}55.9%{opacity:0;top:inherit}}@keyframes self-building-square-spinner{0%{opacity:0}5%,50.9%{opacity:1;top:0}55.9%{opacity:0;top:inherit}}"]
                    }] }
        ];
        /** @nocollapse */
        SelfBuildingSquareSpinnerComponent.ctorParameters = function () { return []; };
        SelfBuildingSquareSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            size: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return SelfBuildingSquareSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var SelfBuildingSquareSpinnerModule = /** @class */ (function () {
        function SelfBuildingSquareSpinnerModule() {
        }
        SelfBuildingSquareSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [SelfBuildingSquareSpinnerComponent],
                        declarations: [SelfBuildingSquareSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return SelfBuildingSquareSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var SemipolarSpinnerComponent = /** @class */ (function () {
        function SemipolarSpinnerComponent() {
            this.animationDuration = 2000;
            this.size = 65;
            this.color = '#fff';
            this.ringsNum = 5;
        }
        Object.defineProperty(SemipolarSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SemipolarSpinnerComponent.prototype, "ringStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    animationDuration: this.animationDuration + "ms",
                    borderTopColor: this.color,
                    borderLeftColor: this.color
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SemipolarSpinnerComponent.prototype, "ringsStyles", {
            get: /**
             * @return {?}
             */ function () {
                /** @type {?} */
                var ringsStyles = [];
                /** @type {?} */
                var delayModifier = 0.1;
                /** @type {?} */
                var ringWidth = this.size * 0.05;
                /** @type {?} */
                var positionIncrement = ringWidth * 2;
                /** @type {?} */
                var sizeDecrement = this.size * 0.2;
                for (var i = 0; i < this.ringsNum; i++) {
                    /** @type {?} */
                    var computedSize = this.size - sizeDecrement * i + "px";
                    /** @type {?} */
                    var computedPosition = positionIncrement * i + "px";
                    /** @type {?} */
                    var style = __assign({ animationDelay: this.animationDuration * delayModifier * (this.ringsNum - i - 1) + "ms", height: computedSize, width: computedSize, left: computedPosition, top: computedPosition, borderWidth: ringWidth + "px" }, this.ringStyle);
                    ringsStyles.push(style);
                }
                return ringsStyles;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        SemipolarSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        SemipolarSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-semipolar-spinner',
                        template: "\n    <div class=\"semipolar-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"ring\" *ngFor=\"let rs of ringsStyles\" [ngStyle]=\"rs\"></div>\n    </div>\n  ",
                        styles: [".semipolar-spinner,.semipolar-spinner *{box-sizing:border-box}.semipolar-spinner{height:65px;width:65px;position:relative}.semipolar-spinner .ring{border-radius:50%;position:absolute;border:solid transparent;border-top-color:#ff1d5e;border-left-color:#ff1d5e;-webkit-animation:2s infinite semipolar-spinner-animation;animation:2s infinite semipolar-spinner-animation}.semipolar-spinner .ring:nth-child(1){height:calc(65px - 65px * .2 * 0);width:calc(65px - 65px * .2 * 0);top:calc(65px * .1 * 0);left:calc(65px * .1 * 0);-webkit-animation-delay:calc(2000ms * .1 * 4);animation-delay:calc(2000ms * .1 * 4);z-index:5}.semipolar-spinner .ring:nth-child(2){height:calc(65px - 65px * .2 * 1);width:calc(65px - 65px * .2 * 1);top:calc(65px * .1 * 1);left:calc(65px * .1 * 1);-webkit-animation-delay:calc(2000ms * .1 * 3);animation-delay:calc(2000ms * .1 * 3);z-index:4}.semipolar-spinner .ring:nth-child(3){height:calc(65px - 65px * .2 * 2);width:calc(65px - 65px * .2 * 2);top:calc(65px * .1 * 2);left:calc(65px * .1 * 2);-webkit-animation-delay:calc(2000ms * .1 * 2);animation-delay:calc(2000ms * .1 * 2);z-index:3}.semipolar-spinner .ring:nth-child(4){height:calc(65px - 65px * .2 * 3);width:calc(65px - 65px * .2 * 3);top:calc(65px * .1 * 3);left:calc(65px * .1 * 3);-webkit-animation-delay:calc(2000ms * .1 * 1);animation-delay:calc(2000ms * .1 * 1);z-index:2}.semipolar-spinner .ring:nth-child(5){height:calc(65px - 65px * .2 * 4);width:calc(65px - 65px * .2 * 4);top:calc(65px * .1 * 4);left:calc(65px * .1 * 4);-webkit-animation-delay:calc(2000ms * .1 * 0);animation-delay:calc(2000ms * .1 * 0);z-index:1}@-webkit-keyframes semipolar-spinner-animation{50%{-webkit-transform:rotate(360deg) scale(.7);transform:rotate(360deg) scale(.7)}}@keyframes semipolar-spinner-animation{50%{-webkit-transform:rotate(360deg) scale(.7);transform:rotate(360deg) scale(.7)}}"]
                    }] }
        ];
        /** @nocollapse */
        SemipolarSpinnerComponent.ctorParameters = function () { return []; };
        SemipolarSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            size: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return SemipolarSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var SemipolarSpinnerModule = /** @class */ (function () {
        function SemipolarSpinnerModule() {
        }
        SemipolarSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [SemipolarSpinnerComponent],
                        declarations: [SemipolarSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return SemipolarSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var SpringSpinnerComponent = /** @class */ (function () {
        function SpringSpinnerComponent() {
            this.size = 60;
            this.color = '#fff';
            this.animationDuration = 2000;
            this.animationBaseName = 'spring-spinner-animation';
            this.currentAnimationName = '';
        }
        Object.defineProperty(SpringSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SpringSpinnerComponent.prototype, "spinnerPartStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size / 2 + "px",
                    width: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SpringSpinnerComponent.prototype, "rotatorStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px",
                    borderRightColor: this.color,
                    borderTopColor: this.color,
                    borderWidth: this.size / 7 + "px",
                    animationDuration: this.animationDuration + "ms",
                    animationName: this.currentAnimationName
                };
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        SpringSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        SpringSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-spring-spinner',
                        template: "\n    <div class=\"spring-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"spring-spinner-part top\" [ngStyle]=\"spinnerPartStyle\">\n        <div class=\"spring-spinner-rotator\" [ngStyle]=\"rotatorStyle\"></div>\n      </div>\n      <div class=\"spring-spinner-part bottom\" [ngStyle]=\"spinnerPartStyle\">\n        <div class=\"spring-spinner-rotator\" [ngStyle]=\"rotatorStyle\"></div>\n      </div>\n    </div>\n  ",
                        styles: [".spring-spinner,.spring-spinner *{box-sizing:border-box}.spring-spinner{height:60px;width:60px}.spring-spinner .spring-spinner-part{overflow:hidden;height:calc(60px / 2);width:60px}.spring-spinner .spring-spinner-part.bottom{-webkit-transform:rotate(180deg) scale(-1,1);transform:rotate(180deg) scale(-1,1)}.spring-spinner .spring-spinner-rotator{width:60px;height:60px;border:solid transparent;border-right-color:#ff1d5e;border-top-color:#ff1d5e;border-radius:50%;box-sizing:border-box;-webkit-animation:3s ease-in-out infinite spring-spinner-animation;animation:3s ease-in-out infinite spring-spinner-animation;-webkit-transform:rotate(-200deg);transform:rotate(-200deg)}@-webkit-keyframes spring-spinner-animation{0%,100%{border-width:calc(60px / 7)}25%,75%{border-width:calc(60px / 23.33)}50%{-webkit-transform:rotate(115deg);transform:rotate(115deg);border-width:calc(60px / 7)}}@keyframes spring-spinner-animation{0%,100%{border-width:calc(60px / 7)}25%,75%{border-width:calc(60px / 23.33)}50%{-webkit-transform:rotate(115deg);transform:rotate(115deg);border-width:calc(60px / 7)}}"]
                    }] }
        ];
        /** @nocollapse */
        SpringSpinnerComponent.ctorParameters = function () { return []; };
        SpringSpinnerComponent.propDecorators = {
            size: [{ type: core.Input }],
            color: [{ type: core.Input }],
            animationDuration: [{ type: core.Input }]
        };
        return SpringSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var SpringSpinnerModule = /** @class */ (function () {
        function SpringSpinnerModule() {
        }
        SpringSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [SpringSpinnerComponent],
                        declarations: [SpringSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return SpringSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var SwappingSquaresSpinnerComponent = /** @class */ (function () {
        function SwappingSquaresSpinnerComponent() {
            this.animationDuration = 1000;
            this.size = 65;
            this.color = '#fff';
            this.animationBaseName = 'swapping-square-animation-child';
            this.squaresNum = 4;
        }
        Object.defineProperty(SwappingSquaresSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SwappingSquaresSpinnerComponent.prototype, "squareStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size * 0.25 / 1.3 + "px",
                    width: this.size * 0.25 / 1.3 + "px",
                    animationDuration: this.animationDuration + "ms",
                    borderWidth: this.size * 0.04 / 1.3 + "px",
                    borderColor: this.color
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SwappingSquaresSpinnerComponent.prototype, "squaresStyles", {
            get: /**
             * @return {?}
             */ function () {
                /** @type {?} */
                var squaresStyles = [];
                /** @type {?} */
                var delay = this.animationDuration * 0.5;
                for (var i = 1; i <= this.squaresNum; i++) {
                    squaresStyles.push(__assign({ animationDelay: (i % 2 === 0 ? delay : 0) + "ms" }, this.squareStyle));
                }
                return squaresStyles;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        SwappingSquaresSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        SwappingSquaresSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-swapping-squares-spinner',
                        template: "\n    <div class=\"swapping-squares-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"square\"\n           *ngFor=\"let ss of squaresStyles; let i = index;\"\n           [ngClass]=\"'square-' + (i + 1)\"\n           [ngStyle]=\"ss\">\n      </div>\n    </div>\n  ",
                        styles: [".swapping-squares-spinner,.swapping-squares-spinner *{box-sizing:border-box}.swapping-squares-spinner{height:65px;width:65px;position:relative;display:flex;flex-direction:row;justify-content:center;align-items:center}.swapping-squares-spinner .square{height:calc(65px * .25 / 1.3);width:calc(65px * .25 / 1.3);-webkit-animation-duration:1s;animation-duration:1s;border:solid #ff1d5e;margin-right:auto;margin-left:auto;position:absolute;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite}.swapping-squares-spinner .square:nth-child(1){-webkit-animation-name:swapping-squares-animation-child-1;animation-name:swapping-squares-animation-child-1;-webkit-animation-delay:.5s;animation-delay:.5s}.swapping-squares-spinner .square:nth-child(2){-webkit-animation-name:swapping-squares-animation-child-2;animation-name:swapping-squares-animation-child-2;-webkit-animation-delay:0s;animation-delay:0s}.swapping-squares-spinner .square:nth-child(3){-webkit-animation-name:swapping-squares-animation-child-3;animation-name:swapping-squares-animation-child-3;-webkit-animation-delay:.5s;animation-delay:.5s}.swapping-squares-spinner .square:nth-child(4){-webkit-animation-name:swapping-squares-animation-child-4;animation-name:swapping-squares-animation-child-4;-webkit-animation-delay:0s;animation-delay:0s}@-webkit-keyframes swapping-squares-animation-child-1{50%{-webkit-transform:translate(150%,150%) scale(2,2);transform:translate(150%,150%) scale(2,2)}}@keyframes swapping-squares-animation-child-1{50%{-webkit-transform:translate(150%,150%) scale(2,2);transform:translate(150%,150%) scale(2,2)}}@-webkit-keyframes swapping-squares-animation-child-2{50%{-webkit-transform:translate(-150%,150%) scale(2,2);transform:translate(-150%,150%) scale(2,2)}}@keyframes swapping-squares-animation-child-2{50%{-webkit-transform:translate(-150%,150%) scale(2,2);transform:translate(-150%,150%) scale(2,2)}}@-webkit-keyframes swapping-squares-animation-child-3{50%{-webkit-transform:translate(-150%,-150%) scale(2,2);transform:translate(-150%,-150%) scale(2,2)}}@keyframes swapping-squares-animation-child-3{50%{-webkit-transform:translate(-150%,-150%) scale(2,2);transform:translate(-150%,-150%) scale(2,2)}}@-webkit-keyframes swapping-squares-animation-child-4{50%{-webkit-transform:translate(150%,-150%) scale(2,2);transform:translate(150%,-150%) scale(2,2)}}@keyframes swapping-squares-animation-child-4{50%{-webkit-transform:translate(150%,-150%) scale(2,2);transform:translate(150%,-150%) scale(2,2)}}"]
                    }] }
        ];
        /** @nocollapse */
        SwappingSquaresSpinnerComponent.ctorParameters = function () { return []; };
        SwappingSquaresSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            size: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return SwappingSquaresSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var SwappingSquaresSpinnerModule = /** @class */ (function () {
        function SwappingSquaresSpinnerModule() {
        }
        SwappingSquaresSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [SwappingSquaresSpinnerComponent],
                        declarations: [SwappingSquaresSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return SwappingSquaresSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var TrinityRingsSpinnerComponent = /** @class */ (function () {
        function TrinityRingsSpinnerComponent() {
            this.containerPadding = 3;
            this.animationDuration = 1500;
            this.size = 60;
            this.color = '#fff';
        }
        Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "outerRingSize", {
            get: /**
             * @return {?}
             */ function () {
                return this.size - this.containerPadding * 2;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "spinnerStyle", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.size + "px",
                    width: this.size + "px",
                    padding: this.containerPadding + "px"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "ring1Style", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.outerRingSize + "px",
                    width: this.outerRingSize + "px",
                    borderColor: this.color,
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "ring2Style", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.outerRingSize * 0.65 + "px",
                    width: this.outerRingSize * 0.65 + "px",
                    borderColor: this.color,
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "ring3Style", {
            get: /**
             * @return {?}
             */ function () {
                return {
                    height: this.outerRingSize * 0.1 + "px",
                    width: this.outerRingSize * 0.1 + "px",
                    borderColor: this.color,
                    animationDuration: this.animationDuration + "ms"
                };
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        TrinityRingsSpinnerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        TrinityRingsSpinnerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-trinity-rings-spinner',
                        template: "\n    <div class=\"trinity-rings-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle circle1\" [ngStyle]=\"ring1Style\"></div>\n      <div class=\"circle circle3\" [ngStyle]=\"ring3Style\"></div>\n      <div class=\"circle circle2\" [ngStyle]=\"ring2Style\"></div>\n    </div>\n  ",
                        styles: [".trinity-rings-spinner,.trinity-rings-spinner *{box-sizing:border-box}.trinity-rings-spinner{height:66px;width:66px;padding:3px;position:relative;display:flex;justify-content:center;align-items:center;flex-direction:row;overflow:hidden;box-sizing:border-box}.trinity-rings-spinner .circle{position:absolute;display:block;border-radius:50%;border:3px solid #ff1d5e;opacity:1}.trinity-rings-spinner .circle:nth-child(1){height:60px;width:60px;-webkit-animation:1.5s linear infinite trinity-rings-spinner-circle1-animation;animation:1.5s linear infinite trinity-rings-spinner-circle1-animation;border-width:3px}.trinity-rings-spinner .circle:nth-child(2){height:calc(60px * .65);width:calc(60px * .65);-webkit-animation:1.5s linear infinite trinity-rings-spinner-circle2-animation;animation:1.5s linear infinite trinity-rings-spinner-circle2-animation;border-width:2px}.trinity-rings-spinner .circle:nth-child(3){height:calc(60px * .1);width:calc(60px * .1);-webkit-animation:1.5s linear infinite trinity-rings-spinner-circle3-animation;animation:1.5s linear infinite trinity-rings-spinner-circle3-animation;border-width:1px}@-webkit-keyframes trinity-rings-spinner-circle1-animation{0%{-webkit-transform:rotateZ(20deg) rotateY(0);transform:rotateZ(20deg) rotateY(0)}100%{-webkit-transform:rotateZ(100deg) rotateY(360deg);transform:rotateZ(100deg) rotateY(360deg)}}@keyframes trinity-rings-spinner-circle1-animation{0%{-webkit-transform:rotateZ(20deg) rotateY(0);transform:rotateZ(20deg) rotateY(0)}100%{-webkit-transform:rotateZ(100deg) rotateY(360deg);transform:rotateZ(100deg) rotateY(360deg)}}@-webkit-keyframes trinity-rings-spinner-circle2-animation{0%{-webkit-transform:rotateZ(100deg) rotateX(0);transform:rotateZ(100deg) rotateX(0)}100%{-webkit-transform:rotateZ(0) rotateX(360deg);transform:rotateZ(0) rotateX(360deg)}}@keyframes trinity-rings-spinner-circle2-animation{0%{-webkit-transform:rotateZ(100deg) rotateX(0);transform:rotateZ(100deg) rotateX(0)}100%{-webkit-transform:rotateZ(0) rotateX(360deg);transform:rotateZ(0) rotateX(360deg)}}@-webkit-keyframes trinity-rings-spinner-circle3-animation{0%{-webkit-transform:rotateZ(100deg) rotateX(-360deg);transform:rotateZ(100deg) rotateX(-360deg)}100%{-webkit-transform:rotateZ(-360deg) rotateX(360deg);transform:rotateZ(-360deg) rotateX(360deg)}}@keyframes trinity-rings-spinner-circle3-animation{0%{-webkit-transform:rotateZ(100deg) rotateX(-360deg);transform:rotateZ(100deg) rotateX(-360deg)}100%{-webkit-transform:rotateZ(-360deg) rotateX(360deg);transform:rotateZ(-360deg) rotateX(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        TrinityRingsSpinnerComponent.ctorParameters = function () { return []; };
        TrinityRingsSpinnerComponent.propDecorators = {
            animationDuration: [{ type: core.Input }],
            size: [{ type: core.Input }],
            color: [{ type: core.Input }]
        };
        return TrinityRingsSpinnerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var TrinityRingsSpinnerModule = /** @class */ (function () {
        function TrinityRingsSpinnerModule() {
        }
        TrinityRingsSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [common.CommonModule],
                        exports: [TrinityRingsSpinnerComponent],
                        declarations: [TrinityRingsSpinnerComponent],
                        providers: [],
                    },] }
        ];
        return TrinityRingsSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */

    exports.AtomSpinnerModule = AtomSpinnerModule;
    exports.BreedingRhombusSpinnerModule = BreedingRhombusSpinnerModule;
    exports.CirclesToRhumbusesSpinnerModule = CirclesToRhumbusesSpinnerModule;
    exports.FingerprintSpinnerModule = FingerprintSpinnerModule;
    exports.FulfillingBouncingCircleSpinnerModule = FulfillingBouncingCircleSpinnerModule;
    exports.FlowerSpinnerModule = FlowerSpinnerModule;
    exports.FulfillingSquareSpinnerModule = FulfillingSquareSpinnerModule;
    exports.HalfCircleSpinnerModule = HalfCircleSpinnerModule;
    exports.HollowDotsSpinnerModule = HollowDotsSpinnerModule;
    exports.IntersectingCirclesSpinnerModule = IntersectingCirclesSpinnerModule;
    exports.LoopingRhumbusesSpinnerModule = LoopingRhumbusesSpinnerModule;
    exports.OrbitSpinnerModule = OrbitSpinnerModule;
    exports.PixelSpinnerModule = PixelSpinnerModule;
    exports.RadarSpinnerModule = RadarSpinnerModule;
    exports.ScalingSquaresSpinnerModule = ScalingSquaresSpinnerModule;
    exports.SelfBuildingSquareSpinnerModule = SelfBuildingSquareSpinnerModule;
    exports.SemipolarSpinnerModule = SemipolarSpinnerModule;
    exports.SpringSpinnerModule = SpringSpinnerModule;
    exports.SwappingSquaresSpinnerModule = SwappingSquaresSpinnerModule;
    exports.TrinityRingsSpinnerModule = TrinityRingsSpinnerModule;
    exports.a = AtomSpinnerComponent;
    exports.b = BreedingRhombusSpinnerComponent;
    exports.c = CirclesToRhumbusesSpinnerComponent;
    exports.d = FingerprintSpinnerComponent;
    exports.f = FlowerSpinnerComponent;
    exports.e = FulfillingBouncingCircleSpinnerComponent;
    exports.g = FulfillingSquareSpinnerComponent;
    exports.h = HalfCircleSpinnerComponent;
    exports.i = HollowDotsSpinnerComponent;
    exports.j = IntersectingCirclesSpinnerComponent;
    exports.k = LoopingRhumbusesSpinnerComponent;
    exports.l = OrbitSpinnerComponent;
    exports.m = PixelSpinnerComponent;
    exports.n = RadarSpinnerComponent;
    exports.o = ScalingSquaresSpinnerComponent;
    exports.p = SelfBuildingSquareSpinnerComponent;
    exports.q = SemipolarSpinnerComponent;
    exports.r = SpringSpinnerComponent;
    exports.s = SwappingSquaresSpinnerComponent;
    exports.t = TrinityRingsSpinnerComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
