// angular-epic-spinners/esm2015/src/app/semipolar-spinner/semipolar-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class SemipolarSpinnerComponent {
    constructor() {
        this.animationDuration = 2000;
        this.size = 65;
        this.color = '#fff';
        this.ringsNum = 5;
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`
        };
    }
    /**
     * @return {?}
     */
    get ringStyle() {
        return {
            animationDuration: `${this.animationDuration}ms`,
            borderTopColor: this.color,
            borderLeftColor: this.color
        };
    }
    /**
     * @return {?}
     */
    get ringsStyles() {
        /** @type {?} */
        const ringsStyles = [];
        /** @type {?} */
        const delayModifier = 0.1;
        /** @type {?} */
        const ringWidth = this.size * 0.05;
        /** @type {?} */
        const positionIncrement = ringWidth * 2;
        /** @type {?} */
        const sizeDecrement = this.size * 0.2;
        for (let i = 0; i < this.ringsNum; i++) {
            /** @type {?} */
            const computedSize = `${this.size - sizeDecrement * i}px`;
            /** @type {?} */
            const computedPosition = `${positionIncrement * i}px`;
            /** @type {?} */
            const style = Object.assign({ animationDelay: `${this.animationDuration * delayModifier * (this.ringsNum - i - 1)}ms`, height: computedSize, width: computedSize, left: computedPosition, top: computedPosition, borderWidth: `${ringWidth}px` }, this.ringStyle);
            ringsStyles.push(style);
        }
        return ringsStyles;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
SemipolarSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-semipolar-spinner',
                template: `
    <div class="semipolar-spinner" [ngStyle]="spinnerStyle">
      <div class="ring" *ngFor="let rs of ringsStyles" [ngStyle]="rs"></div>
    </div>
  `,
                styles: [".semipolar-spinner,.semipolar-spinner *{box-sizing:border-box}.semipolar-spinner{height:65px;width:65px;position:relative}.semipolar-spinner .ring{border-radius:50%;position:absolute;border:solid transparent;border-top-color:#ff1d5e;border-left-color:#ff1d5e;-webkit-animation:2s infinite semipolar-spinner-animation;animation:2s infinite semipolar-spinner-animation}.semipolar-spinner .ring:nth-child(1){height:calc(65px - 65px * .2 * 0);width:calc(65px - 65px * .2 * 0);top:calc(65px * .1 * 0);left:calc(65px * .1 * 0);-webkit-animation-delay:calc(2000ms * .1 * 4);animation-delay:calc(2000ms * .1 * 4);z-index:5}.semipolar-spinner .ring:nth-child(2){height:calc(65px - 65px * .2 * 1);width:calc(65px - 65px * .2 * 1);top:calc(65px * .1 * 1);left:calc(65px * .1 * 1);-webkit-animation-delay:calc(2000ms * .1 * 3);animation-delay:calc(2000ms * .1 * 3);z-index:4}.semipolar-spinner .ring:nth-child(3){height:calc(65px - 65px * .2 * 2);width:calc(65px - 65px * .2 * 2);top:calc(65px * .1 * 2);left:calc(65px * .1 * 2);-webkit-animation-delay:calc(2000ms * .1 * 2);animation-delay:calc(2000ms * .1 * 2);z-index:3}.semipolar-spinner .ring:nth-child(4){height:calc(65px - 65px * .2 * 3);width:calc(65px - 65px * .2 * 3);top:calc(65px * .1 * 3);left:calc(65px * .1 * 3);-webkit-animation-delay:calc(2000ms * .1 * 1);animation-delay:calc(2000ms * .1 * 1);z-index:2}.semipolar-spinner .ring:nth-child(5){height:calc(65px - 65px * .2 * 4);width:calc(65px - 65px * .2 * 4);top:calc(65px * .1 * 4);left:calc(65px * .1 * 4);-webkit-animation-delay:calc(2000ms * .1 * 0);animation-delay:calc(2000ms * .1 * 0);z-index:1}@-webkit-keyframes semipolar-spinner-animation{50%{-webkit-transform:rotate(360deg) scale(.7);transform:rotate(360deg) scale(.7)}}@keyframes semipolar-spinner-animation{50%{-webkit-transform:rotate(360deg) scale(.7);transform:rotate(360deg) scale(.7)}}"]
            }] }
];
/** @nocollapse */
SemipolarSpinnerComponent.ctorParameters = () => [];
SemipolarSpinnerComponent.propDecorators = {
    animationDuration: [{ type: Input }],
    size: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    SemipolarSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    SemipolarSpinnerComponent.prototype.size;
    /** @type {?} */
    SemipolarSpinnerComponent.prototype.color;
    /** @type {?} */
    SemipolarSpinnerComponent.prototype.ringsNum;
}
