// angular-epic-spinners/esm2015/src/app/fingerprint-spinner/fingerprint-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class FingerprintSpinnerComponent {
    constructor() {
        this.size = 60;
        this.animationDuration = 1500;
        this.color = '#fff';
        this.ringsNum = 9;
        this.containerPadding = 2;
    }
    /**
     * @return {?}
     */
    get outerRingSize() {
        return this.size - this.containerPadding * 2;
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`,
            padding: `${this.containerPadding}px`
        };
    }
    /**
     * @return {?}
     */
    get ringStyle() {
        return {
            borderTopColor: this.color,
            animationDuration: `${this.animationDuration}ms`
        };
    }
    /**
     * @return {?}
     */
    get ringsStyles() {
        /** @type {?} */
        const ringsStyles = [];
        /** @type {?} */
        const ringBase = this.outerRingSize / (this.ringsNum);
        /** @type {?} */
        const ringInc = ringBase;
        for (let i = 1; i <= this.ringsNum; i++) {
            ringsStyles.push(Object.assign({}, this.ringStyle, { animationDelay: `${i * 50}ms`, height: `${ringBase + (i + 1) * ringInc}px`, width: `${ringBase + (i + 1) * ringInc}px` }));
        }
        return ringsStyles;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
FingerprintSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fingerprint-spinner',
                template: `
    <div class="fingerprint-spinner" [ngStyle]="spinnerStyle">
      <div class="spinner-ring" *ngFor="let rs of ringsStyles" [ngStyle]="rs"></div>
    </div>

  `,
                styles: [".fingerprint-spinner,.fingerprint-spinner *{box-sizing:border-box}.fingerprint-spinner{height:64px;width:64px;padding:2px;overflow:hidden;position:relative}.fingerprint-spinner .spinner-ring{position:absolute;border-radius:50%;border:2px solid transparent;border-top-color:#ff1d5e;-webkit-animation:1.5s cubic-bezier(.68,-.75,.265,1.75) infinite forwards fingerprint-spinner-animation;animation:1.5s cubic-bezier(.68,-.75,.265,1.75) infinite forwards fingerprint-spinner-animation;margin:auto;bottom:0;left:0;right:0;top:0}.fingerprint-spinner .spinner-ring:nth-child(1){height:calc(60px / 9 + 0 * 60px / 9);width:calc(60px / 9 + 0 * 60px / 9);-webkit-animation-delay:calc(50ms * 1);animation-delay:calc(50ms * 1)}.fingerprint-spinner .spinner-ring:nth-child(2){height:calc(60px / 9 + 1 * 60px / 9);width:calc(60px / 9 + 1 * 60px / 9);-webkit-animation-delay:calc(50ms * 2);animation-delay:calc(50ms * 2)}.fingerprint-spinner .spinner-ring:nth-child(3){height:calc(60px / 9 + 2 * 60px / 9);width:calc(60px / 9 + 2 * 60px / 9);-webkit-animation-delay:calc(50ms * 3);animation-delay:calc(50ms * 3)}.fingerprint-spinner .spinner-ring:nth-child(4){height:calc(60px / 9 + 3 * 60px / 9);width:calc(60px / 9 + 3 * 60px / 9);-webkit-animation-delay:calc(50ms * 4);animation-delay:calc(50ms * 4)}.fingerprint-spinner .spinner-ring:nth-child(5){height:calc(60px / 9 + 4 * 60px / 9);width:calc(60px / 9 + 4 * 60px / 9);-webkit-animation-delay:calc(50ms * 5);animation-delay:calc(50ms * 5)}.fingerprint-spinner .spinner-ring:nth-child(6){height:calc(60px / 9 + 5 * 60px / 9);width:calc(60px / 9 + 5 * 60px / 9);-webkit-animation-delay:calc(50ms * 6);animation-delay:calc(50ms * 6)}.fingerprint-spinner .spinner-ring:nth-child(7){height:calc(60px / 9 + 6 * 60px / 9);width:calc(60px / 9 + 6 * 60px / 9);-webkit-animation-delay:calc(50ms * 7);animation-delay:calc(50ms * 7)}.fingerprint-spinner .spinner-ring:nth-child(8){height:calc(60px / 9 + 7 * 60px / 9);width:calc(60px / 9 + 7 * 60px / 9);-webkit-animation-delay:calc(50ms * 8);animation-delay:calc(50ms * 8)}.fingerprint-spinner .spinner-ring:nth-child(9){height:calc(60px / 9 + 8 * 60px / 9);width:calc(60px / 9 + 8 * 60px / 9);-webkit-animation-delay:calc(50ms * 9);animation-delay:calc(50ms * 9)}@-webkit-keyframes fingerprint-spinner-animation{100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes fingerprint-spinner-animation{100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
            }] }
];
/** @nocollapse */
FingerprintSpinnerComponent.ctorParameters = () => [];
FingerprintSpinnerComponent.propDecorators = {
    size: [{ type: Input }],
    animationDuration: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    FingerprintSpinnerComponent.prototype.size;
    /** @type {?} */
    FingerprintSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    FingerprintSpinnerComponent.prototype.color;
    /** @type {?} */
    FingerprintSpinnerComponent.prototype.ringsNum;
    /** @type {?} */
    FingerprintSpinnerComponent.prototype.containerPadding;
}
