// angular-epic-spinners/esm5/public_api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
export { AtomSpinnerModule } from './src/app/atom-spinner/atom-spinner.module';
export { BreedingRhombusSpinnerModule } from './src/app/breeding-rhombus-spinner/breeding-rhombus-spinner.module';
export { CirclesToRhumbusesSpinnerModule } from './src/app/circles-to-rhumbuses-spinner/circles-to-rhumbuses-spinner.module';
export { FingerprintSpinnerModule } from './src/app/fingerprint-spinner/fingerprint-spinner.module';
export { FulfillingBouncingCircleSpinnerModule } from './src/app/fulfilling-bouncing-circle-spinner/fulfilling-bouncing-circle-spinner.module';
export { FlowerSpinnerModule } from './src/app/flower-spinner/flower-spinner.module';
export { FulfillingSquareSpinnerModule } from './src/app/fulfilling-square-spinner/fulfilling-square-spinner.module';
export { HalfCircleSpinnerModule } from './src/app/half-circle-spinner/half-circle-spinner.module';
export { HollowDotsSpinnerModule } from './src/app/hollow-dots-spinner/hollow-dots-spinner.module';
export { IntersectingCirclesSpinnerModule } from './src/app/intersecting-circles-spinner/intersecting-circles-spinner.module';
export { LoopingRhumbusesSpinnerModule } from './src/app/looping-rhombuses-spinner/looping-rhumbuses-spinner.module';
export { OrbitSpinnerModule } from './src/app/orbit-spinner/orbit-spinner.module';
export { PixelSpinnerModule } from './src/app/pixel-spinner/pixel-spinner.module';
export { RadarSpinnerModule } from './src/app/radar-spinner/radar-spinner.module';
export { ScalingSquaresSpinnerModule } from './src/app/scaling-squares-spinner/scaling-squares-spinner.module';
export { SelfBuildingSquareSpinnerModule } from './src/app/self-building-square-spinner/self-building-square-spinner.module';
export { SemipolarSpinnerModule } from './src/app/semipolar-spinner/semipolar-spinner.module';
export { SpringSpinnerModule } from './src/app/spring-spinner/spring-spinner.module';
export { SwappingSquaresSpinnerModule } from './src/app/swapping-squares-spinner/swapping-squares-spinner.module';
export { TrinityRingsSpinnerModule } from './src/app/trinity-rings-spinner/trinity-rings-spinner.module';
