// angular-epic-spinners/esm5/src/app/intersecting-circles-spinner/intersecting-circles-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
var IntersectingCirclesSpinnerComponent = /** @class */ (function () {
    function IntersectingCirclesSpinnerComponent() {
        this.animationDuration = 1200;
        this.size = 70;
        this.color = '#fff';
    }
    Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "circleSize", {
        get: /**
         * @return {?}
         */
        function () {
            return this.size / 2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                width: this.size + "px",
                height: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "spinnerBlockStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                animationDuration: this.animationDuration + "ms",
                width: this.circleSize + "px",
                height: this.circleSize + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "circleStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                borderColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "circleStyles", {
        get: /**
         * @return {?}
         */
        function () {
            var _this = this;
            /** @type {?} */
            var circlesPositions = [
                { top: 0, left: 0 },
                { left: this.circleSize * -0.36 + "px", top: this.circleSize * 0.2 + "px" },
                { left: this.circleSize * -0.36 + "px", top: this.circleSize * -0.2 + "px" },
                { left: 0, top: this.circleSize * -0.36 + "px" },
                { left: this.circleSize * 0.36 + "px", top: this.circleSize * -0.2 + "px" },
                { left: this.circleSize * 0.36 + "px", top: this.circleSize * 0.2 + "px" },
                { left: 0, top: this.circleSize * 0.36 + "px" }
            ];
            return circlesPositions.map(function (cp) {
                return tslib_1.__assign({ cp: cp }, _this.circleStyle);
            });
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    IntersectingCirclesSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    IntersectingCirclesSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-intersecting-circles-spinner',
                    template: "\n    <div class=\"intersecting-circles-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"spinnerBlock\" [ngStyle]=\"spinnerBlockStyle\">\n        <span class=\"circle\" *ngFor=\"let cs of circleStyles\" [ngStyle]=\"cs\"></span>\n      </div>\n    </div>\n  ",
                    styles: [".intersecting-circles-spinner,.intersecting-circles-spinner *{box-sizing:border-box}.intersecting-circles-spinner{height:70px;width:70px;position:relative;display:flex;flex-direction:row;justify-content:center;align-items:center}.intersecting-circles-spinner .spinnerBlock{-webkit-animation:1.2s linear infinite intersecting-circles-spinners-animation;animation:1.2s linear infinite intersecting-circles-spinners-animation;-webkit-transform-origin:center;transform-origin:center;display:block;height:35px;width:35px}.intersecting-circles-spinner .circle{display:block;border:2px solid #ff1d5e;border-radius:50%;height:100%;width:100%;position:absolute;left:0;top:0}.intersecting-circles-spinner .circle:nth-child(1){left:0;top:0}.intersecting-circles-spinner .circle:nth-child(2){left:calc(35px * -.36);top:calc(35px * .2)}.intersecting-circles-spinner .circle:nth-child(3){left:calc(35px * -.36);top:calc(35px * -.2)}.intersecting-circles-spinner .circle:nth-child(4){left:0;top:calc(35px * -.36)}.intersecting-circles-spinner .circle:nth-child(5){left:calc(35px * .36);top:calc(35px * -.2)}.intersecting-circles-spinner .circle:nth-child(6){left:calc(35px * .36);top:calc(35px * .2)}.intersecting-circles-spinner .circle:nth-child(7){left:0;top:calc(35px * .36)}@-webkit-keyframes intersecting-circles-spinners-animation{from{-webkit-transform:rotate(0);transform:rotate(0)}to{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes intersecting-circles-spinners-animation{from{-webkit-transform:rotate(0);transform:rotate(0)}to{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
                }] }
    ];
    /** @nocollapse */
    IntersectingCirclesSpinnerComponent.ctorParameters = function () { return []; };
    IntersectingCirclesSpinnerComponent.propDecorators = {
        animationDuration: [{ type: Input }],
        size: [{ type: Input }],
        color: [{ type: Input }]
    };
    return IntersectingCirclesSpinnerComponent;
}());
export { IntersectingCirclesSpinnerComponent };
if (false) {
    /** @type {?} */
    IntersectingCirclesSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    IntersectingCirclesSpinnerComponent.prototype.size;
    /** @type {?} */
    IntersectingCirclesSpinnerComponent.prototype.color;
}
