// angular-epic-spinners/esm2015/src/app/radar-spinner/radar-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class RadarSpinnerComponent {
    constructor() {
        this.size = 110;
        this.animationDuration = 2000;
        this.color = '#fff';
        this.circlesNum = 4;
    }
    /**
     * @return {?}
     */
    get borderWidth() {
        return this.size * 5 / 110;
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`
        };
    }
    /**
     * @return {?}
     */
    get circleStyle() {
        return {
            animationDuration: `${this.animationDuration}ms`
        };
    }
    /**
     * @return {?}
     */
    get circleInnerContainerStyle() {
        return {
            borderWidth: `${this.borderWidth}px`
        };
    }
    /**
     * @return {?}
     */
    get circleInnerStyle() {
        return {
            borderLeftColor: this.color,
            borderRightColor: this.color,
            borderWidth: `${this.borderWidth}px`
        };
    }
    /**
     * @return {?}
     */
    get circlesStyles() {
        /** @type {?} */
        const circlesStyles = [];
        /** @type {?} */
        const delay = this.animationDuration * 0.15;
        for (let i = 0; i < this.circlesNum; i++) {
            circlesStyles.push(Object.assign({
                padding: `${this.borderWidth * 2 * i}px`,
                animationDelay: `${i === this.circlesNum - 1 ? 0 : delay}ms`
            }, this.circleStyle));
        }
        return circlesStyles;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
RadarSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-radar-spinner',
                template: `
    <div class="radar-spinner" [ngStyle]="spinnerStyle">
      <div class="circle" *ngFor="let cs of circlesStyles" [ngStyle]="cs">
        <div class="circle-inner-container" [ngStyle]="circleInnerContainerStyle">
          <div class="circle-inner" [ngStyle]="circleInnerStyle"></div>
        </div>
      </div>
    </div>
  `,
                styles: [".radar-spinner,.radar-spinner *{box-sizing:border-box}.radar-spinner{height:60px;width:60px;position:relative}.radar-spinner .circle{position:absolute;height:100%;width:100%;top:0;left:0;-webkit-animation:2s infinite radar-spinner-animation;animation:2s infinite radar-spinner-animation}.radar-spinner .circle:nth-child(1){padding:calc(60px * 5 * 2 * 0 / 110);-webkit-animation-delay:.3s;animation-delay:.3s}.radar-spinner .circle:nth-child(2){padding:calc(60px * 5 * 2 * 1 / 110);-webkit-animation-delay:.3s;animation-delay:.3s}.radar-spinner .circle:nth-child(3){padding:calc(60px * 5 * 2 * 2 / 110);-webkit-animation-delay:.3s;animation-delay:.3s}.radar-spinner .circle:nth-child(4){padding:calc(60px * 5 * 2 * 3 / 110);-webkit-animation-delay:0s;animation-delay:0s}.radar-spinner .circle-inner,.radar-spinner .circle-inner-container{height:100%;width:100%;border-radius:50%;border:solid transparent}.radar-spinner .circle-inner{border-left-color:#ff1d5e;border-right-color:#ff1d5e}@-webkit-keyframes radar-spinner-animation{50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}100%{-webkit-transform:rotate(0);transform:rotate(0)}}@keyframes radar-spinner-animation{50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}100%{-webkit-transform:rotate(0);transform:rotate(0)}}"]
            }] }
];
/** @nocollapse */
RadarSpinnerComponent.ctorParameters = () => [];
RadarSpinnerComponent.propDecorators = {
    size: [{ type: Input }],
    animationDuration: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    RadarSpinnerComponent.prototype.size;
    /** @type {?} */
    RadarSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    RadarSpinnerComponent.prototype.color;
    /** @type {?} */
    RadarSpinnerComponent.prototype.circlesNum;
}
