// angular-epic-spinners/esm5/src/app/half-circle-spinner/half-circle-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
var HalfCircleSpinnerComponent = /** @class */ (function () {
    function HalfCircleSpinnerComponent() {
        this.animationDuration = 1000;
        this.size = 60;
        this.color = '#fff';
    }
    Object.defineProperty(HalfCircleSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HalfCircleSpinnerComponent.prototype, "circleStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                borderWidth: this.size / 10 + "px",
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HalfCircleSpinnerComponent.prototype, "circle1Style", {
        get: /**
         * @return {?}
         */
        function () {
            return tslib_1.__assign({}, this.circleStyle, { borderTopColor: this.color });
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HalfCircleSpinnerComponent.prototype, "circle2Style", {
        get: /**
         * @return {?}
         */
        function () {
            return tslib_1.__assign({}, this.circleStyle, { borderBottomColor: this.color });
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    HalfCircleSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    HalfCircleSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-half-circle-spinner',
                    template: "\n    <div class=\"half-circle-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle circle-1\" [ngStyle]=\"circle1Style\"></div>\n      <div class=\"circle circle-2\" [ngStyle]=\"circle2Style\"></div>\n    </div>\n  ",
                    styles: [".half-circle-spinner,.half-circle-spinner *{box-sizing:border-box}.half-circle-spinner{width:60px;height:60px;border-radius:100%;position:relative}.half-circle-spinner .circle{content:\"\";position:absolute;width:100%;height:100%;border-radius:100%;border:solid transparent}.half-circle-spinner .circle.circle-1{border-top-color:#ff1d5e;-webkit-animation:1s infinite half-circle-spinner-animation;animation:1s infinite half-circle-spinner-animation}.half-circle-spinner .circle.circle-2{border-bottom-color:#ff1d5e;-webkit-animation:1s infinite alternate half-circle-spinner-animation;animation:1s infinite alternate half-circle-spinner-animation}@-webkit-keyframes half-circle-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes half-circle-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
                }] }
    ];
    /** @nocollapse */
    HalfCircleSpinnerComponent.ctorParameters = function () { return []; };
    HalfCircleSpinnerComponent.propDecorators = {
        animationDuration: [{ type: Input }],
        size: [{ type: Input }],
        color: [{ type: Input }]
    };
    return HalfCircleSpinnerComponent;
}());
export { HalfCircleSpinnerComponent };
if (false) {
    /** @type {?} */
    HalfCircleSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    HalfCircleSpinnerComponent.prototype.size;
    /** @type {?} */
    HalfCircleSpinnerComponent.prototype.color;
}
