// angular-epic-spinners/esm2015/src/app/orbit-spinner/orbit-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class OrbitSpinnerComponent {
    constructor() {
        this.animationDuration = 1000;
        this.size = 50;
        this.color = '#fff';
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`
        };
    }
    /**
     * @return {?}
     */
    get orbitStyle() {
        return {
            borderColor: this.color,
            animationDuration: `${this.animationDuration}ms`
        };
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
OrbitSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-orbit-spinner',
                template: `
    <div class="orbit-spinner" [ngStyle]="spinnerStyle">
      <div class="orbit one" [ngStyle]="orbitStyle"></div>
      <div class="orbit two" [ngStyle]="orbitStyle"></div>
      <div class="orbit three" [ngStyle]="orbitStyle"></div>
    </div>
  `,
                styles: [".orbit-spinner,.orbit-spinner *{box-sizing:border-box}.orbit-spinner{height:55px;width:55px;border-radius:50%;-webkit-perspective:800px;perspective:800px}.orbit-spinner .orbit{position:absolute;box-sizing:border-box;width:100%;height:100%;border-radius:50%}.orbit-spinner .orbit:nth-child(1){left:0;top:0;-webkit-animation:1.2s linear infinite orbit-spinner-orbit-one-animation;animation:1.2s linear infinite orbit-spinner-orbit-one-animation;border-bottom:3px solid #ff1d5e}.orbit-spinner .orbit:nth-child(2){right:0;top:0;-webkit-animation:1.2s linear infinite orbit-spinner-orbit-two-animation;animation:1.2s linear infinite orbit-spinner-orbit-two-animation;border-right:3px solid #ff1d5e}.orbit-spinner .orbit:nth-child(3){right:0;bottom:0;-webkit-animation:1.2s linear infinite orbit-spinner-orbit-three-animation;animation:1.2s linear infinite orbit-spinner-orbit-three-animation;border-top:3px solid #ff1d5e}@-webkit-keyframes orbit-spinner-orbit-one-animation{0%{-webkit-transform:rotateX(35deg) rotateY(-45deg) rotateZ(0);transform:rotateX(35deg) rotateY(-45deg) rotateZ(0)}100%{-webkit-transform:rotateX(35deg) rotateY(-45deg) rotateZ(360deg);transform:rotateX(35deg) rotateY(-45deg) rotateZ(360deg)}}@keyframes orbit-spinner-orbit-one-animation{0%{-webkit-transform:rotateX(35deg) rotateY(-45deg) rotateZ(0);transform:rotateX(35deg) rotateY(-45deg) rotateZ(0)}100%{-webkit-transform:rotateX(35deg) rotateY(-45deg) rotateZ(360deg);transform:rotateX(35deg) rotateY(-45deg) rotateZ(360deg)}}@-webkit-keyframes orbit-spinner-orbit-two-animation{0%{-webkit-transform:rotateX(50deg) rotateY(10deg) rotateZ(0);transform:rotateX(50deg) rotateY(10deg) rotateZ(0)}100%{-webkit-transform:rotateX(50deg) rotateY(10deg) rotateZ(360deg);transform:rotateX(50deg) rotateY(10deg) rotateZ(360deg)}}@keyframes orbit-spinner-orbit-two-animation{0%{-webkit-transform:rotateX(50deg) rotateY(10deg) rotateZ(0);transform:rotateX(50deg) rotateY(10deg) rotateZ(0)}100%{-webkit-transform:rotateX(50deg) rotateY(10deg) rotateZ(360deg);transform:rotateX(50deg) rotateY(10deg) rotateZ(360deg)}}@-webkit-keyframes orbit-spinner-orbit-three-animation{0%{-webkit-transform:rotateX(35deg) rotateY(55deg) rotateZ(0);transform:rotateX(35deg) rotateY(55deg) rotateZ(0)}100%{-webkit-transform:rotateX(35deg) rotateY(55deg) rotateZ(360deg);transform:rotateX(35deg) rotateY(55deg) rotateZ(360deg)}}@keyframes orbit-spinner-orbit-three-animation{0%{-webkit-transform:rotateX(35deg) rotateY(55deg) rotateZ(0);transform:rotateX(35deg) rotateY(55deg) rotateZ(0)}100%{-webkit-transform:rotateX(35deg) rotateY(55deg) rotateZ(360deg);transform:rotateX(35deg) rotateY(55deg) rotateZ(360deg)}}"]
            }] }
];
/** @nocollapse */
OrbitSpinnerComponent.ctorParameters = () => [];
OrbitSpinnerComponent.propDecorators = {
    animationDuration: [{ type: Input }],
    size: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    OrbitSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    OrbitSpinnerComponent.prototype.size;
    /** @type {?} */
    OrbitSpinnerComponent.prototype.color;
}
