// angular-epic-spinners/esm5/src/app/fingerprint-spinner/fingerprint-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
var FingerprintSpinnerComponent = /** @class */ (function () {
    function FingerprintSpinnerComponent() {
        this.size = 60;
        this.animationDuration = 1500;
        this.color = '#fff';
        this.ringsNum = 9;
        this.containerPadding = 2;
    }
    Object.defineProperty(FingerprintSpinnerComponent.prototype, "outerRingSize", {
        get: /**
         * @return {?}
         */
        function () {
            return this.size - this.containerPadding * 2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FingerprintSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                padding: this.containerPadding + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FingerprintSpinnerComponent.prototype, "ringStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                borderTopColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FingerprintSpinnerComponent.prototype, "ringsStyles", {
        get: /**
         * @return {?}
         */
        function () {
            /** @type {?} */
            var ringsStyles = [];
            /** @type {?} */
            var ringBase = this.outerRingSize / (this.ringsNum);
            /** @type {?} */
            var ringInc = ringBase;
            for (var i = 1; i <= this.ringsNum; i++) {
                ringsStyles.push(tslib_1.__assign({}, this.ringStyle, { animationDelay: i * 50 + "ms", height: ringBase + (i + 1) * ringInc + "px", width: ringBase + (i + 1) * ringInc + "px" }));
            }
            return ringsStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    FingerprintSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    FingerprintSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-fingerprint-spinner',
                    template: "\n    <div class=\"fingerprint-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"spinner-ring\" *ngFor=\"let rs of ringsStyles\" [ngStyle]=\"rs\"></div>\n    </div>\n\n  ",
                    styles: [".fingerprint-spinner,.fingerprint-spinner *{box-sizing:border-box}.fingerprint-spinner{height:64px;width:64px;padding:2px;overflow:hidden;position:relative}.fingerprint-spinner .spinner-ring{position:absolute;border-radius:50%;border:2px solid transparent;border-top-color:#ff1d5e;-webkit-animation:1.5s cubic-bezier(.68,-.75,.265,1.75) infinite forwards fingerprint-spinner-animation;animation:1.5s cubic-bezier(.68,-.75,.265,1.75) infinite forwards fingerprint-spinner-animation;margin:auto;bottom:0;left:0;right:0;top:0}.fingerprint-spinner .spinner-ring:nth-child(1){height:calc(60px / 9 + 0 * 60px / 9);width:calc(60px / 9 + 0 * 60px / 9);-webkit-animation-delay:calc(50ms * 1);animation-delay:calc(50ms * 1)}.fingerprint-spinner .spinner-ring:nth-child(2){height:calc(60px / 9 + 1 * 60px / 9);width:calc(60px / 9 + 1 * 60px / 9);-webkit-animation-delay:calc(50ms * 2);animation-delay:calc(50ms * 2)}.fingerprint-spinner .spinner-ring:nth-child(3){height:calc(60px / 9 + 2 * 60px / 9);width:calc(60px / 9 + 2 * 60px / 9);-webkit-animation-delay:calc(50ms * 3);animation-delay:calc(50ms * 3)}.fingerprint-spinner .spinner-ring:nth-child(4){height:calc(60px / 9 + 3 * 60px / 9);width:calc(60px / 9 + 3 * 60px / 9);-webkit-animation-delay:calc(50ms * 4);animation-delay:calc(50ms * 4)}.fingerprint-spinner .spinner-ring:nth-child(5){height:calc(60px / 9 + 4 * 60px / 9);width:calc(60px / 9 + 4 * 60px / 9);-webkit-animation-delay:calc(50ms * 5);animation-delay:calc(50ms * 5)}.fingerprint-spinner .spinner-ring:nth-child(6){height:calc(60px / 9 + 5 * 60px / 9);width:calc(60px / 9 + 5 * 60px / 9);-webkit-animation-delay:calc(50ms * 6);animation-delay:calc(50ms * 6)}.fingerprint-spinner .spinner-ring:nth-child(7){height:calc(60px / 9 + 6 * 60px / 9);width:calc(60px / 9 + 6 * 60px / 9);-webkit-animation-delay:calc(50ms * 7);animation-delay:calc(50ms * 7)}.fingerprint-spinner .spinner-ring:nth-child(8){height:calc(60px / 9 + 7 * 60px / 9);width:calc(60px / 9 + 7 * 60px / 9);-webkit-animation-delay:calc(50ms * 8);animation-delay:calc(50ms * 8)}.fingerprint-spinner .spinner-ring:nth-child(9){height:calc(60px / 9 + 8 * 60px / 9);width:calc(60px / 9 + 8 * 60px / 9);-webkit-animation-delay:calc(50ms * 9);animation-delay:calc(50ms * 9)}@-webkit-keyframes fingerprint-spinner-animation{100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes fingerprint-spinner-animation{100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
                }] }
    ];
    /** @nocollapse */
    FingerprintSpinnerComponent.ctorParameters = function () { return []; };
    FingerprintSpinnerComponent.propDecorators = {
        size: [{ type: Input }],
        animationDuration: [{ type: Input }],
        color: [{ type: Input }]
    };
    return FingerprintSpinnerComponent;
}());
export { FingerprintSpinnerComponent };
if (false) {
    /** @type {?} */
    FingerprintSpinnerComponent.prototype.size;
    /** @type {?} */
    FingerprintSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    FingerprintSpinnerComponent.prototype.color;
    /** @type {?} */
    FingerprintSpinnerComponent.prototype.ringsNum;
    /** @type {?} */
    FingerprintSpinnerComponent.prototype.containerPadding;
}
