// angular-epic-spinners/esm5/src/app/swapping-squares-spinner/swapping-squares-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
var SwappingSquaresSpinnerComponent = /** @class */ (function () {
    function SwappingSquaresSpinnerComponent() {
        this.animationDuration = 1000;
        this.size = 65;
        this.color = '#fff';
        this.animationBaseName = 'swapping-square-animation-child';
        this.squaresNum = 4;
    }
    Object.defineProperty(SwappingSquaresSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SwappingSquaresSpinnerComponent.prototype, "squareStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.size * 0.25 / 1.3 + "px",
                width: this.size * 0.25 / 1.3 + "px",
                animationDuration: this.animationDuration + "ms",
                borderWidth: this.size * 0.04 / 1.3 + "px",
                borderColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SwappingSquaresSpinnerComponent.prototype, "squaresStyles", {
        get: /**
         * @return {?}
         */
        function () {
            /** @type {?} */
            var squaresStyles = [];
            /** @type {?} */
            var delay = this.animationDuration * 0.5;
            for (var i = 1; i <= this.squaresNum; i++) {
                squaresStyles.push(tslib_1.__assign({ animationDelay: (i % 2 === 0 ? delay : 0) + "ms" }, this.squareStyle));
            }
            return squaresStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    SwappingSquaresSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    SwappingSquaresSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-swapping-squares-spinner',
                    template: "\n    <div class=\"swapping-squares-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"square\"\n           *ngFor=\"let ss of squaresStyles; let i = index;\"\n           [ngClass]=\"'square-' + (i + 1)\"\n           [ngStyle]=\"ss\">\n      </div>\n    </div>\n  ",
                    styles: [".swapping-squares-spinner,.swapping-squares-spinner *{box-sizing:border-box}.swapping-squares-spinner{height:65px;width:65px;position:relative;display:flex;flex-direction:row;justify-content:center;align-items:center}.swapping-squares-spinner .square{height:calc(65px * .25 / 1.3);width:calc(65px * .25 / 1.3);-webkit-animation-duration:1s;animation-duration:1s;border:solid #ff1d5e;margin-right:auto;margin-left:auto;position:absolute;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite}.swapping-squares-spinner .square:nth-child(1){-webkit-animation-name:swapping-squares-animation-child-1;animation-name:swapping-squares-animation-child-1;-webkit-animation-delay:.5s;animation-delay:.5s}.swapping-squares-spinner .square:nth-child(2){-webkit-animation-name:swapping-squares-animation-child-2;animation-name:swapping-squares-animation-child-2;-webkit-animation-delay:0s;animation-delay:0s}.swapping-squares-spinner .square:nth-child(3){-webkit-animation-name:swapping-squares-animation-child-3;animation-name:swapping-squares-animation-child-3;-webkit-animation-delay:.5s;animation-delay:.5s}.swapping-squares-spinner .square:nth-child(4){-webkit-animation-name:swapping-squares-animation-child-4;animation-name:swapping-squares-animation-child-4;-webkit-animation-delay:0s;animation-delay:0s}@-webkit-keyframes swapping-squares-animation-child-1{50%{-webkit-transform:translate(150%,150%) scale(2,2);transform:translate(150%,150%) scale(2,2)}}@keyframes swapping-squares-animation-child-1{50%{-webkit-transform:translate(150%,150%) scale(2,2);transform:translate(150%,150%) scale(2,2)}}@-webkit-keyframes swapping-squares-animation-child-2{50%{-webkit-transform:translate(-150%,150%) scale(2,2);transform:translate(-150%,150%) scale(2,2)}}@keyframes swapping-squares-animation-child-2{50%{-webkit-transform:translate(-150%,150%) scale(2,2);transform:translate(-150%,150%) scale(2,2)}}@-webkit-keyframes swapping-squares-animation-child-3{50%{-webkit-transform:translate(-150%,-150%) scale(2,2);transform:translate(-150%,-150%) scale(2,2)}}@keyframes swapping-squares-animation-child-3{50%{-webkit-transform:translate(-150%,-150%) scale(2,2);transform:translate(-150%,-150%) scale(2,2)}}@-webkit-keyframes swapping-squares-animation-child-4{50%{-webkit-transform:translate(150%,-150%) scale(2,2);transform:translate(150%,-150%) scale(2,2)}}@keyframes swapping-squares-animation-child-4{50%{-webkit-transform:translate(150%,-150%) scale(2,2);transform:translate(150%,-150%) scale(2,2)}}"]
                }] }
    ];
    /** @nocollapse */
    SwappingSquaresSpinnerComponent.ctorParameters = function () { return []; };
    SwappingSquaresSpinnerComponent.propDecorators = {
        animationDuration: [{ type: Input }],
        size: [{ type: Input }],
        color: [{ type: Input }]
    };
    return SwappingSquaresSpinnerComponent;
}());
export { SwappingSquaresSpinnerComponent };
if (false) {
    /** @type {?} */
    SwappingSquaresSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    SwappingSquaresSpinnerComponent.prototype.size;
    /** @type {?} */
    SwappingSquaresSpinnerComponent.prototype.color;
    /** @type {?} */
    SwappingSquaresSpinnerComponent.prototype.animationBaseName;
    /** @type {?} */
    SwappingSquaresSpinnerComponent.prototype.squaresNum;
}
