// angular-epic-spinners/esm2015/src/app/fulfilling-bouncing-circle-spinner/fulfilling-bouncing-circle-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class FulfillingBouncingCircleSpinnerComponent {
    constructor() {
        this.size = 60;
        this.animationDuration = 4000;
        this.color = '#fff';
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`,
            animationDuration: `${this.animationDuration}ms`
        };
    }
    /**
     * @return {?}
     */
    get orbitStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`,
            borderColor: this.color,
            borderWidth: `${this.size * 0.03}px`,
            animationDuration: `${this.animationDuration}ms`
        };
    }
    /**
     * @return {?}
     */
    get circleStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`,
            borderColor: this.color,
            color: this.color,
            borderWidth: `${this.size * 0.1}px`,
            animationDuration: `${this.animationDuration}ms`
        };
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
FulfillingBouncingCircleSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fulfilling-bouncing-circle-spinner',
                template: `
    <div class="fulfilling-bouncing-circle-spinner" [ngStyle]="spinnerStyle">
      <div class="circle" [ngStyle]="circleStyle"></div>
      <div class="orbit" [ngStyle]="orbitStyle"></div>
    </div>
  `,
                styles: [".fulfilling-bouncing-circle-spinner,.fulfilling-bouncing-circle-spinner *{box-sizing:border-box}.fulfilling-bouncing-circle-spinner{height:60px;width:60px;position:relative;-webkit-animation:4s infinite fulfilling-bouncing-circle-spinner-animation;animation:4s infinite fulfilling-bouncing-circle-spinner-animation}.fulfilling-bouncing-circle-spinner .orbit{height:60px;width:60px;position:absolute;top:0;left:0;border-radius:50%;border:solid #ff1d5e;-webkit-animation:4s infinite fulfilling-bouncing-circle-spinner-orbit-animation;animation:4s infinite fulfilling-bouncing-circle-spinner-orbit-animation}.fulfilling-bouncing-circle-spinner .circle{height:60px;width:60px;color:#ff1d5e;display:block;border-radius:50%;position:relative;border:solid #ff1d5e;-webkit-animation:4s infinite fulfilling-bouncing-circle-spinner-circle-animation;animation:4s infinite fulfilling-bouncing-circle-spinner-circle-animation;-webkit-transform:rotate(0) scale(1);transform:rotate(0) scale(1)}@-webkit-keyframes fulfilling-bouncing-circle-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes fulfilling-bouncing-circle-spinner-animation{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@-webkit-keyframes fulfilling-bouncing-circle-spinner-orbit-animation{0%,50%{-webkit-transform:scale(1);transform:scale(1)}62.5%{-webkit-transform:scale(.8);transform:scale(.8)}75%{-webkit-transform:scale(1);transform:scale(1)}87.5%{-webkit-transform:scale(.8);transform:scale(.8)}100%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes fulfilling-bouncing-circle-spinner-orbit-animation{0%,50%{-webkit-transform:scale(1);transform:scale(1)}62.5%{-webkit-transform:scale(.8);transform:scale(.8)}75%{-webkit-transform:scale(1);transform:scale(1)}87.5%{-webkit-transform:scale(.8);transform:scale(.8)}100%{-webkit-transform:scale(1);transform:scale(1)}}@-webkit-keyframes fulfilling-bouncing-circle-spinner-circle-animation{0%{-webkit-transform:scale(1);transform:scale(1);border-color:transparent;border-top-color:inherit}16.7%{border-color:initial initial transparent transparent}33.4%{border-color:transparent;border-top-color:inherit;border-right-color:inherit;border-bottom-color:inherit}50%{border-color:inherit;-webkit-transform:scale(1);transform:scale(1)}62.5%{border-color:inherit;-webkit-transform:scale(1.4);transform:scale(1.4)}75%{border-color:inherit;-webkit-transform:scale(1);transform:scale(1);opacity:1}87.5%{border-color:inherit;-webkit-transform:scale(1.4);transform:scale(1.4)}100%{border-color:transparent;border-top-color:inherit;-webkit-transform:scale(1);transform:scale(1)}}@keyframes fulfilling-bouncing-circle-spinner-circle-animation{0%{-webkit-transform:scale(1);transform:scale(1);border-color:transparent;border-top-color:inherit}16.7%{border-color:initial initial transparent transparent}33.4%{border-color:transparent;border-top-color:inherit;border-right-color:inherit;border-bottom-color:inherit}50%{border-color:inherit;-webkit-transform:scale(1);transform:scale(1)}62.5%{border-color:inherit;-webkit-transform:scale(1.4);transform:scale(1.4)}75%{border-color:inherit;-webkit-transform:scale(1);transform:scale(1);opacity:1}87.5%{border-color:inherit;-webkit-transform:scale(1.4);transform:scale(1.4)}100%{border-color:transparent;border-top-color:inherit;-webkit-transform:scale(1);transform:scale(1)}}"]
            }] }
];
/** @nocollapse */
FulfillingBouncingCircleSpinnerComponent.ctorParameters = () => [];
FulfillingBouncingCircleSpinnerComponent.propDecorators = {
    size: [{ type: Input }],
    animationDuration: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    FulfillingBouncingCircleSpinnerComponent.prototype.size;
    /** @type {?} */
    FulfillingBouncingCircleSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    FulfillingBouncingCircleSpinnerComponent.prototype.color;
}
