// angular-epic-spinners/esm5/src/app/atom-spinner/atom-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var AtomSpinnerComponent = /** @class */ (function () {
    function AtomSpinnerComponent() {
        this.size = 60;
        this.animationDuration = 1000;
        this.color = 'red';
    }
    Object.defineProperty(AtomSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AtomSpinnerComponent.prototype, "circleStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                color: this.color,
                fontSize: this.size * 0.24 + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AtomSpinnerComponent.prototype, "lineStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                animationDuration: this.animationDuration + "ms",
                borderLeftWidth: this.size / 25 + "px",
                borderRightWidth: this.size / 25 + "px",
                borderLeftColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    AtomSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    AtomSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-atom-spinner',
                    template: "\n      <div class=\"atom-spinner\" [ngStyle]=\"spinnerStyle\">\n        <div class=\"spinner-inner\">\n          <div class=\"spinner-line\" [ngStyle]=\"lineStyle\"></div>\n          <div class=\"spinner-line\" [ngStyle]=\"lineStyle\"></div>\n          <div class=\"spinner-line\" [ngStyle]=\"lineStyle\"></div>\n          <div class=\"spinner-circle\" [ngStyle]=\"circleStyle\">\n            &#9679;\n          </div>\n        </div>\n      </div>\n  ",
                    styles: [".atom-spinner,.atom-spinner *{box-sizing:border-box}.atom-spinner{height:60px;width:60px;overflow:hidden}.atom-spinner .spinner-inner{position:relative;display:block;height:100%;width:100%}.atom-spinner .spinner-circle{display:block;position:absolute;color:#ff1d5e;font-size:calc(60px * .24);top:50%;left:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%)}.atom-spinner .spinner-line{position:absolute;width:100%;height:100%;border-radius:50%;-webkit-animation-duration:1s;animation-duration:1s;border-left-width:calc(60px / 25);border-top-width:calc(60px / 25);border-left-color:#ff1d5e;border-left-style:solid;border-top-style:solid;border-top-color:transparent}.atom-spinner .spinner-line:nth-child(1){-webkit-animation:1s linear infinite atom-spinner-animation-1;animation:1s linear infinite atom-spinner-animation-1;-webkit-transform:rotateZ(120deg) rotateX(66deg) rotateZ(0);transform:rotateZ(120deg) rotateX(66deg) rotateZ(0)}.atom-spinner .spinner-line:nth-child(2){-webkit-animation:1s linear infinite atom-spinner-animation-2;animation:1s linear infinite atom-spinner-animation-2;-webkit-transform:rotateZ(240deg) rotateX(66deg) rotateZ(0);transform:rotateZ(240deg) rotateX(66deg) rotateZ(0)}.atom-spinner .spinner-line:nth-child(3){-webkit-animation:1s linear infinite atom-spinner-animation-3;animation:1s linear infinite atom-spinner-animation-3;-webkit-transform:rotateZ(360deg) rotateX(66deg) rotateZ(0);transform:rotateZ(360deg) rotateX(66deg) rotateZ(0)}@-webkit-keyframes atom-spinner-animation-1{100%{-webkit-transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg)}}@keyframes atom-spinner-animation-1{100%{-webkit-transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg)}}@-webkit-keyframes atom-spinner-animation-2{100%{-webkit-transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg)}}@keyframes atom-spinner-animation-2{100%{-webkit-transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg)}}@-webkit-keyframes atom-spinner-animation-3{100%{-webkit-transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg)}}@keyframes atom-spinner-animation-3{100%{-webkit-transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg)}}"]
                }] }
    ];
    /** @nocollapse */
    AtomSpinnerComponent.ctorParameters = function () { return []; };
    AtomSpinnerComponent.propDecorators = {
        size: [{ type: Input }],
        animationDuration: [{ type: Input }],
        color: [{ type: Input }]
    };
    return AtomSpinnerComponent;
}());
export { AtomSpinnerComponent };
if (false) {
    /** @type {?} */
    AtomSpinnerComponent.prototype.size;
    /** @type {?} */
    AtomSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    AtomSpinnerComponent.prototype.color;
}
