// angular-epic-spinners/esm5/src/app/self-building-square-spinner/self-building-square-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
var SelfBuildingSquareSpinnerComponent = /** @class */ (function () {
    function SelfBuildingSquareSpinnerComponent() {
        this.animationDuration = 6000;
        this.size = 40;
        this.color = '#fff';
        this.squaresNum = 9;
    }
    Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "squareSize", {
        get: /**
         * @return {?}
         */
        function () {
            return this.size / 4;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "initialTopPosition", {
        get: /**
         * @return {?}
         */
        function () {
            return -this.squareSize * 2 / 3;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                top: -this.initialTopPosition + "px",
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "squareStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.squareSize + "px",
                width: this.squareSize + "px",
                top: this.initialTopPosition + "px",
                marginRight: this.squareSize / 3 + "px",
                marginTop: this.squareSize / 3 + "px",
                animationDuration: this.animationDuration + "ms",
                background: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "squaresStyles", {
        get: /**
         * @return {?}
         */
        function () {
            /** @type {?} */
            var squaresStyles = [];
            /** @type {?} */
            var delaysMultipliers = [6, 7, 8, 3, 4, 5, 0, 1, 2];
            /** @type {?} */
            var delayModifier = this.animationDuration * 0.05;
            for (var i = 0; i < this.squaresNum; i++) {
                squaresStyles.push(tslib_1.__assign({ animationDelay: delayModifier * delaysMultipliers[i] + "ms" }, this.squareStyle));
            }
            return squaresStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    SelfBuildingSquareSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    SelfBuildingSquareSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-self-building-square-spinner',
                    template: "\n    <div class=\"self-building-square-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"square\"\n           *ngFor=\"let ss of squaresStyles; let i = index;\"\n           [ngStyle]=\"ss\"\n           [ngClass]=\"{'clear': i && i % 3 === 0}\"\n      >\n      </div>\n    </div>\n  ",
                    styles: [".self-building-square-spinner,.self-building-square-spinner *{box-sizing:border-box}.self-building-square-spinner{height:40px;width:40px;top:calc(-10px * 2 / 3)}.self-building-square-spinner .square{height:10px;width:10px;top:calc(-10px * 2 / 3);margin-right:calc(10px / 3);margin-top:calc(10px / 3);background:#ff1d5e;float:left;position:relative;opacity:0;-webkit-animation:6s infinite self-building-square-spinner;animation:6s infinite self-building-square-spinner}.self-building-square-spinner .square:nth-child(1){-webkit-animation-delay:calc(300ms * 6);animation-delay:calc(300ms * 6)}.self-building-square-spinner .square:nth-child(2){-webkit-animation-delay:calc(300ms * 7);animation-delay:calc(300ms * 7)}.self-building-square-spinner .square:nth-child(3){-webkit-animation-delay:calc(300ms * 8);animation-delay:calc(300ms * 8)}.self-building-square-spinner .square:nth-child(4){-webkit-animation-delay:calc(300ms * 3);animation-delay:calc(300ms * 3)}.self-building-square-spinner .square:nth-child(5){-webkit-animation-delay:calc(300ms * 4);animation-delay:calc(300ms * 4)}.self-building-square-spinner .square:nth-child(6){-webkit-animation-delay:calc(300ms * 5);animation-delay:calc(300ms * 5)}.self-building-square-spinner .square:nth-child(7){-webkit-animation-delay:calc(300ms * 0);animation-delay:calc(300ms * 0)}.self-building-square-spinner .square:nth-child(8){-webkit-animation-delay:calc(300ms * 1);animation-delay:calc(300ms * 1)}.self-building-square-spinner .square:nth-child(9){-webkit-animation-delay:calc(300ms * 2);animation-delay:calc(300ms * 2)}.self-building-square-spinner .clear{clear:both}@-webkit-keyframes self-building-square-spinner{0%{opacity:0}5%,50.9%{opacity:1;top:0}55.9%{opacity:0;top:inherit}}@keyframes self-building-square-spinner{0%{opacity:0}5%,50.9%{opacity:1;top:0}55.9%{opacity:0;top:inherit}}"]
                }] }
    ];
    /** @nocollapse */
    SelfBuildingSquareSpinnerComponent.ctorParameters = function () { return []; };
    SelfBuildingSquareSpinnerComponent.propDecorators = {
        animationDuration: [{ type: Input }],
        size: [{ type: Input }],
        color: [{ type: Input }]
    };
    return SelfBuildingSquareSpinnerComponent;
}());
export { SelfBuildingSquareSpinnerComponent };
if (false) {
    /** @type {?} */
    SelfBuildingSquareSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    SelfBuildingSquareSpinnerComponent.prototype.size;
    /** @type {?} */
    SelfBuildingSquareSpinnerComponent.prototype.color;
    /** @type {?} */
    SelfBuildingSquareSpinnerComponent.prototype.squaresNum;
}
