// angular-epic-spinners/esm5/src/app/semipolar-spinner/semipolar-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
var SemipolarSpinnerComponent = /** @class */ (function () {
    function SemipolarSpinnerComponent() {
        this.animationDuration = 2000;
        this.size = 65;
        this.color = '#fff';
        this.ringsNum = 5;
    }
    Object.defineProperty(SemipolarSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SemipolarSpinnerComponent.prototype, "ringStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                animationDuration: this.animationDuration + "ms",
                borderTopColor: this.color,
                borderLeftColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SemipolarSpinnerComponent.prototype, "ringsStyles", {
        get: /**
         * @return {?}
         */
        function () {
            /** @type {?} */
            var ringsStyles = [];
            /** @type {?} */
            var delayModifier = 0.1;
            /** @type {?} */
            var ringWidth = this.size * 0.05;
            /** @type {?} */
            var positionIncrement = ringWidth * 2;
            /** @type {?} */
            var sizeDecrement = this.size * 0.2;
            for (var i = 0; i < this.ringsNum; i++) {
                /** @type {?} */
                var computedSize = this.size - sizeDecrement * i + "px";
                /** @type {?} */
                var computedPosition = positionIncrement * i + "px";
                /** @type {?} */
                var style = tslib_1.__assign({ animationDelay: this.animationDuration * delayModifier * (this.ringsNum - i - 1) + "ms", height: computedSize, width: computedSize, left: computedPosition, top: computedPosition, borderWidth: ringWidth + "px" }, this.ringStyle);
                ringsStyles.push(style);
            }
            return ringsStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    SemipolarSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    SemipolarSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-semipolar-spinner',
                    template: "\n    <div class=\"semipolar-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"ring\" *ngFor=\"let rs of ringsStyles\" [ngStyle]=\"rs\"></div>\n    </div>\n  ",
                    styles: [".semipolar-spinner,.semipolar-spinner *{box-sizing:border-box}.semipolar-spinner{height:65px;width:65px;position:relative}.semipolar-spinner .ring{border-radius:50%;position:absolute;border:solid transparent;border-top-color:#ff1d5e;border-left-color:#ff1d5e;-webkit-animation:2s infinite semipolar-spinner-animation;animation:2s infinite semipolar-spinner-animation}.semipolar-spinner .ring:nth-child(1){height:calc(65px - 65px * .2 * 0);width:calc(65px - 65px * .2 * 0);top:calc(65px * .1 * 0);left:calc(65px * .1 * 0);-webkit-animation-delay:calc(2000ms * .1 * 4);animation-delay:calc(2000ms * .1 * 4);z-index:5}.semipolar-spinner .ring:nth-child(2){height:calc(65px - 65px * .2 * 1);width:calc(65px - 65px * .2 * 1);top:calc(65px * .1 * 1);left:calc(65px * .1 * 1);-webkit-animation-delay:calc(2000ms * .1 * 3);animation-delay:calc(2000ms * .1 * 3);z-index:4}.semipolar-spinner .ring:nth-child(3){height:calc(65px - 65px * .2 * 2);width:calc(65px - 65px * .2 * 2);top:calc(65px * .1 * 2);left:calc(65px * .1 * 2);-webkit-animation-delay:calc(2000ms * .1 * 2);animation-delay:calc(2000ms * .1 * 2);z-index:3}.semipolar-spinner .ring:nth-child(4){height:calc(65px - 65px * .2 * 3);width:calc(65px - 65px * .2 * 3);top:calc(65px * .1 * 3);left:calc(65px * .1 * 3);-webkit-animation-delay:calc(2000ms * .1 * 1);animation-delay:calc(2000ms * .1 * 1);z-index:2}.semipolar-spinner .ring:nth-child(5){height:calc(65px - 65px * .2 * 4);width:calc(65px - 65px * .2 * 4);top:calc(65px * .1 * 4);left:calc(65px * .1 * 4);-webkit-animation-delay:calc(2000ms * .1 * 0);animation-delay:calc(2000ms * .1 * 0);z-index:1}@-webkit-keyframes semipolar-spinner-animation{50%{-webkit-transform:rotate(360deg) scale(.7);transform:rotate(360deg) scale(.7)}}@keyframes semipolar-spinner-animation{50%{-webkit-transform:rotate(360deg) scale(.7);transform:rotate(360deg) scale(.7)}}"]
                }] }
    ];
    /** @nocollapse */
    SemipolarSpinnerComponent.ctorParameters = function () { return []; };
    SemipolarSpinnerComponent.propDecorators = {
        animationDuration: [{ type: Input }],
        size: [{ type: Input }],
        color: [{ type: Input }]
    };
    return SemipolarSpinnerComponent;
}());
export { SemipolarSpinnerComponent };
if (false) {
    /** @type {?} */
    SemipolarSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    SemipolarSpinnerComponent.prototype.size;
    /** @type {?} */
    SemipolarSpinnerComponent.prototype.color;
    /** @type {?} */
    SemipolarSpinnerComponent.prototype.ringsNum;
}
