// angular-epic-spinners/esm5/src/app/hollow-dots-spinner/hollow-dots-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
var HollowDotsSpinnerComponent = /** @class */ (function () {
    function HollowDotsSpinnerComponent() {
        this.animationDuration = 1000;
        this.dotSize = 15;
        this.dotsNum = 3;
        this.color = '#fff';
    }
    Object.defineProperty(HollowDotsSpinnerComponent.prototype, "horizontalMargin", {
        get: /**
         * @return {?}
         */
        function () {
            return this.dotSize / 2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HollowDotsSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.dotSize + "px",
                width: this.dotSize + "px",
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HollowDotsSpinnerComponent.prototype, "dotStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                width: this.dotSize + "px",
                height: this.dotSize + "px",
                margin: "0 " + this.horizontalMargin + "px",
                borderWidth: this.dotSize / 5 + "px",
                borderColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HollowDotsSpinnerComponent.prototype, "dotsStyles", {
        get: /**
         * @return {?}
         */
        function () {
            /** @type {?} */
            var dotsStyles = [];
            /** @type {?} */
            var delayModifier = 0.3;
            /** @type {?} */
            var basicDelay = 1000;
            for (var i = 1; i <= this.dotsNum; i++) {
                /** @type {?} */
                var style = tslib_1.__assign({}, this.dotStyle, { animationDelay: basicDelay * i * delayModifier + "ms" });
                dotsStyles.push(style);
            }
            return dotsStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    HollowDotsSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    HollowDotsSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-hollow-dots-spinner',
                    template: "\n    <div class=\"hollow-dots-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"dot\" *ngFor=\"let ds of dotsStyles\" [ngStyle]=\"ds\"></div>\n    </div>\n  ",
                    styles: [".hollow-dots-spinner,.hollow-dots-spinner *{box-sizing:border-box}.hollow-dots-spinner{height:15px;width:calc(30px * 3)}.hollow-dots-spinner .dot{width:15px;height:15px;margin:0 calc(15px / 2);border:solid #ff1d5e;border-radius:50%;float:left;-webkit-transform:scale(0);transform:scale(0);-webkit-animation:1s infinite hollow-dots-spinner-animation;animation:1s infinite hollow-dots-spinner-animation}.hollow-dots-spinner .dot:nth-child(1){-webkit-animation-delay:calc(300ms * 1);animation-delay:calc(300ms * 1)}.hollow-dots-spinner .dot:nth-child(2){-webkit-animation-delay:calc(300ms * 2);animation-delay:calc(300ms * 2)}.hollow-dots-spinner .dot:nth-child(3){-webkit-animation-delay:calc(300ms * 3);animation-delay:calc(300ms * 3)}@-webkit-keyframes hollow-dots-spinner-animation{50%{-webkit-transform:scale(1);transform:scale(1);opacity:1}100%{opacity:0}}@keyframes hollow-dots-spinner-animation{50%{-webkit-transform:scale(1);transform:scale(1);opacity:1}100%{opacity:0}}"]
                }] }
    ];
    /** @nocollapse */
    HollowDotsSpinnerComponent.ctorParameters = function () { return []; };
    HollowDotsSpinnerComponent.propDecorators = {
        animationDuration: [{ type: Input }],
        dotSize: [{ type: Input }],
        dotsNum: [{ type: Input }],
        color: [{ type: Input }]
    };
    return HollowDotsSpinnerComponent;
}());
export { HollowDotsSpinnerComponent };
if (false) {
    /** @type {?} */
    HollowDotsSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    HollowDotsSpinnerComponent.prototype.dotSize;
    /** @type {?} */
    HollowDotsSpinnerComponent.prototype.dotsNum;
    /** @type {?} */
    HollowDotsSpinnerComponent.prototype.color;
}
