// angular-epic-spinners/esm2015/src/app/circles-to-rhumbuses-spinner/circles-to-rhumbuses-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class CirclesToRhumbusesSpinnerComponent {
    constructor() {
        this.circleSize = 15;
        this.circlesNum = 3;
        this.animationDuration = 1200;
        this.color = '#fff';
    }
    /**
     * @return {?}
     */
    get circleMarginLeft() {
        return this.circleSize * 1.125;
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.circleSize}px`,
            width: `${(this.circleSize + this.circleMarginLeft) * this.circlesNum}px`
        };
    }
    /**
     * @return {?}
     */
    get circleStyle() {
        return {
            borderColor: this.color,
            animationDuration: `${this.animationDuration}ms`,
            height: `${this.circleSize}px`,
            width: `${this.circleSize}px`,
            marginLeft: `${this.circleMarginLeft}px`
        };
    }
    /**
     * @return {?}
     */
    get circlesStyles() {
        /** @type {?} */
        const circlesStyles = [];
        /** @type {?} */
        const delay = 150;
        for (let i = 1; i <= this.circlesNum; i++) {
            /** @type {?} */
            const style = Object.assign({}, this.circleStyle, { animationDelay: `${i * delay}ms` });
            circlesStyles.push(style);
        }
        return circlesStyles;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
CirclesToRhumbusesSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-circles-to-rhumbuses-spinner',
                template: `
    <div class="circles-to-rhombuses-spinner" [ngStyle]="spinnerStyle">
      <div class="circle" *ngFor="let cs of circlesStyles; let i = index;" [ngStyle]="cs"></div>
    </div>
  `,
                styles: [".circles-to-rhombuses-spinner,.circles-to-rhombuses-spinner *{box-sizing:border-box}.circles-to-rhombuses-spinner{height:15px;width:calc((15px + 15px * 1.125) * 3);display:flex;align-items:center;justify-content:center}.circles-to-rhombuses-spinner .circle{height:15px;width:15px;margin-left:calc(15px * 1.125);-webkit-transform:rotate(45deg);transform:rotate(45deg);border-radius:10%;border:3px solid #ff1d5e;overflow:hidden;background:0 0;-webkit-animation:1.2s linear infinite circles-to-rhombuses-animation;animation:1.2s linear infinite circles-to-rhombuses-animation}.circles-to-rhombuses-spinner .circle:nth-child(1){-webkit-animation-delay:calc(150ms * 1);animation-delay:calc(150ms * 1);margin-left:0}.circles-to-rhombuses-spinner .circle:nth-child(2){-webkit-animation-delay:calc(150ms * 2);animation-delay:calc(150ms * 2)}.circles-to-rhombuses-spinner .circle:nth-child(3){-webkit-animation-delay:calc(150ms * 3);animation-delay:calc(150ms * 3)}@-webkit-keyframes circles-to-rhombuses-animation{0%,17.5%{border-radius:10%}50%{border-radius:100%}100%,93.5%{border-radius:10%}}@keyframes circles-to-rhombuses-animation{0%,17.5%{border-radius:10%}50%{border-radius:100%}100%,93.5%{border-radius:10%}}@-webkit-keyframes circles-to-rhombuses-background-animation{50%{opacity:.4}}@keyframes circles-to-rhombuses-background-animation{50%{opacity:.4}}"]
            }] }
];
/** @nocollapse */
CirclesToRhumbusesSpinnerComponent.ctorParameters = () => [];
CirclesToRhumbusesSpinnerComponent.propDecorators = {
    circleSize: [{ type: Input }],
    circlesNum: [{ type: Input }],
    animationDuration: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    CirclesToRhumbusesSpinnerComponent.prototype.circleSize;
    /** @type {?} */
    CirclesToRhumbusesSpinnerComponent.prototype.circlesNum;
    /** @type {?} */
    CirclesToRhumbusesSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    CirclesToRhumbusesSpinnerComponent.prototype.color;
}
