// angular-epic-spinners/esm2015/src/app/looping-rhombuses-spinner/looping-rhumbuses-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class LoopingRhumbusesSpinnerComponent {
    constructor() {
        this.animationDuration = 2500;
        this.rhombusSize = 15;
        this.color = '#fff';
        this.rhombusesNum = 3;
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.rhombusSize}px`,
            width: `${this.rhombusSize * 4}px`
        };
    }
    /**
     * @return {?}
     */
    get rhombusStyle() {
        return {
            height: `${this.rhombusSize}px`,
            width: `${this.rhombusSize}px`,
            background: this.color,
            animationDuration: `${this.animationDuration}ms`,
            left: `${this.rhombusSize * 4}px`
        };
    }
    /**
     * @return {?}
     */
    get rhombusesStyles() {
        /** @type {?} */
        const rhombusesStyles = [];
        /** @type {?} */
        const delay = -this.animationDuration / 1.5;
        for (let i = 1; i <= this.rhombusesNum; i++) {
            /** @type {?} */
            const style = Object.assign({
                animationDelay: `${i * delay}ms`
            }, this.rhombusStyle);
            rhombusesStyles.push(style);
        }
        return rhombusesStyles;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
LoopingRhumbusesSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-looping-rhombuses-spinner',
                template: `
    <div class="looping-rhombuses-spinner" [ngStyle]="spinnerStyle">
      <div class="rhombus"
           *ngFor="let rs of rhombusesStyles"
           [ngStyle]="rs">
      </div>
    </div>
  `,
                styles: [".looping-rhombuses-spinner,.looping-rhombuses-spinner *{box-sizing:border-box}.looping-rhombuses-spinner{width:calc(15px * 4);height:15px;position:relative}.looping-rhombuses-spinner .rhombus{height:15px;width:15px;background:#ff1d5e;left:calc(15px * 4);position:absolute;margin:0 auto;border-radius:2px;-webkit-transform:translateY(0) rotate(45deg) scale(0);transform:translateY(0) rotate(45deg) scale(0);-webkit-animation:2.5s linear infinite looping-rhombuses-spinner-animation;animation:2.5s linear infinite looping-rhombuses-spinner-animation}.looping-rhombuses-spinner .rhombus:nth-child(1){-webkit-animation-delay:calc(2500ms * 1 / -1.5);animation-delay:calc(2500ms * 1 / -1.5)}.looping-rhombuses-spinner .rhombus:nth-child(2){-webkit-animation-delay:calc(2500ms * 2 / -1.5);animation-delay:calc(2500ms * 2 / -1.5)}.looping-rhombuses-spinner .rhombus:nth-child(3){-webkit-animation-delay:calc(2500ms * 3 / -1.5);animation-delay:calc(2500ms * 3 / -1.5)}@-webkit-keyframes looping-rhombuses-spinner-animation{0%{-webkit-transform:translateX(0) rotate(45deg) scale(0);transform:translateX(0) rotate(45deg) scale(0)}50%{-webkit-transform:translateX(-233%) rotate(45deg) scale(1);transform:translateX(-233%) rotate(45deg) scale(1)}100%{-webkit-transform:translateX(-466%) rotate(45deg) scale(0);transform:translateX(-466%) rotate(45deg) scale(0)}}@keyframes looping-rhombuses-spinner-animation{0%{-webkit-transform:translateX(0) rotate(45deg) scale(0);transform:translateX(0) rotate(45deg) scale(0)}50%{-webkit-transform:translateX(-233%) rotate(45deg) scale(1);transform:translateX(-233%) rotate(45deg) scale(1)}100%{-webkit-transform:translateX(-466%) rotate(45deg) scale(0);transform:translateX(-466%) rotate(45deg) scale(0)}}"]
            }] }
];
/** @nocollapse */
LoopingRhumbusesSpinnerComponent.ctorParameters = () => [];
LoopingRhumbusesSpinnerComponent.propDecorators = {
    animationDuration: [{ type: Input }],
    rhombusSize: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    LoopingRhumbusesSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    LoopingRhumbusesSpinnerComponent.prototype.rhombusSize;
    /** @type {?} */
    LoopingRhumbusesSpinnerComponent.prototype.color;
    /** @type {?} */
    LoopingRhumbusesSpinnerComponent.prototype.rhombusesNum;
}
