// angular-epic-spinners/esm5/src/app/trinity-rings-spinner/trinity-rings-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var TrinityRingsSpinnerComponent = /** @class */ (function () {
    function TrinityRingsSpinnerComponent() {
        this.containerPadding = 3;
        this.animationDuration = 1500;
        this.size = 60;
        this.color = '#fff';
    }
    Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "outerRingSize", {
        get: /**
         * @return {?}
         */
        function () {
            return this.size - this.containerPadding * 2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                padding: this.containerPadding + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "ring1Style", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.outerRingSize + "px",
                width: this.outerRingSize + "px",
                borderColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "ring2Style", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.outerRingSize * 0.65 + "px",
                width: this.outerRingSize * 0.65 + "px",
                borderColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "ring3Style", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.outerRingSize * 0.1 + "px",
                width: this.outerRingSize * 0.1 + "px",
                borderColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    TrinityRingsSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    TrinityRingsSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-trinity-rings-spinner',
                    template: "\n    <div class=\"trinity-rings-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle circle1\" [ngStyle]=\"ring1Style\"></div>\n      <div class=\"circle circle3\" [ngStyle]=\"ring3Style\"></div>\n      <div class=\"circle circle2\" [ngStyle]=\"ring2Style\"></div>\n    </div>\n  ",
                    styles: [".trinity-rings-spinner,.trinity-rings-spinner *{box-sizing:border-box}.trinity-rings-spinner{height:66px;width:66px;padding:3px;position:relative;display:flex;justify-content:center;align-items:center;flex-direction:row;overflow:hidden;box-sizing:border-box}.trinity-rings-spinner .circle{position:absolute;display:block;border-radius:50%;border:3px solid #ff1d5e;opacity:1}.trinity-rings-spinner .circle:nth-child(1){height:60px;width:60px;-webkit-animation:1.5s linear infinite trinity-rings-spinner-circle1-animation;animation:1.5s linear infinite trinity-rings-spinner-circle1-animation;border-width:3px}.trinity-rings-spinner .circle:nth-child(2){height:calc(60px * .65);width:calc(60px * .65);-webkit-animation:1.5s linear infinite trinity-rings-spinner-circle2-animation;animation:1.5s linear infinite trinity-rings-spinner-circle2-animation;border-width:2px}.trinity-rings-spinner .circle:nth-child(3){height:calc(60px * .1);width:calc(60px * .1);-webkit-animation:1.5s linear infinite trinity-rings-spinner-circle3-animation;animation:1.5s linear infinite trinity-rings-spinner-circle3-animation;border-width:1px}@-webkit-keyframes trinity-rings-spinner-circle1-animation{0%{-webkit-transform:rotateZ(20deg) rotateY(0);transform:rotateZ(20deg) rotateY(0)}100%{-webkit-transform:rotateZ(100deg) rotateY(360deg);transform:rotateZ(100deg) rotateY(360deg)}}@keyframes trinity-rings-spinner-circle1-animation{0%{-webkit-transform:rotateZ(20deg) rotateY(0);transform:rotateZ(20deg) rotateY(0)}100%{-webkit-transform:rotateZ(100deg) rotateY(360deg);transform:rotateZ(100deg) rotateY(360deg)}}@-webkit-keyframes trinity-rings-spinner-circle2-animation{0%{-webkit-transform:rotateZ(100deg) rotateX(0);transform:rotateZ(100deg) rotateX(0)}100%{-webkit-transform:rotateZ(0) rotateX(360deg);transform:rotateZ(0) rotateX(360deg)}}@keyframes trinity-rings-spinner-circle2-animation{0%{-webkit-transform:rotateZ(100deg) rotateX(0);transform:rotateZ(100deg) rotateX(0)}100%{-webkit-transform:rotateZ(0) rotateX(360deg);transform:rotateZ(0) rotateX(360deg)}}@-webkit-keyframes trinity-rings-spinner-circle3-animation{0%{-webkit-transform:rotateZ(100deg) rotateX(-360deg);transform:rotateZ(100deg) rotateX(-360deg)}100%{-webkit-transform:rotateZ(-360deg) rotateX(360deg);transform:rotateZ(-360deg) rotateX(360deg)}}@keyframes trinity-rings-spinner-circle3-animation{0%{-webkit-transform:rotateZ(100deg) rotateX(-360deg);transform:rotateZ(100deg) rotateX(-360deg)}100%{-webkit-transform:rotateZ(-360deg) rotateX(360deg);transform:rotateZ(-360deg) rotateX(360deg)}}"]
                }] }
    ];
    /** @nocollapse */
    TrinityRingsSpinnerComponent.ctorParameters = function () { return []; };
    TrinityRingsSpinnerComponent.propDecorators = {
        animationDuration: [{ type: Input }],
        size: [{ type: Input }],
        color: [{ type: Input }]
    };
    return TrinityRingsSpinnerComponent;
}());
export { TrinityRingsSpinnerComponent };
if (false) {
    /** @type {?} */
    TrinityRingsSpinnerComponent.prototype.containerPadding;
    /** @type {?} */
    TrinityRingsSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    TrinityRingsSpinnerComponent.prototype.size;
    /** @type {?} */
    TrinityRingsSpinnerComponent.prototype.color;
}
