// angular-epic-spinners/esm2015/src/app/intersecting-circles-spinner/intersecting-circles-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class IntersectingCirclesSpinnerComponent {
    constructor() {
        this.animationDuration = 1200;
        this.size = 70;
        this.color = '#fff';
    }
    /**
     * @return {?}
     */
    get circleSize() {
        return this.size / 2;
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            width: `${this.size}px`,
            height: `${this.size}px`
        };
    }
    /**
     * @return {?}
     */
    get spinnerBlockStyle() {
        return {
            animationDuration: `${this.animationDuration}ms`,
            width: `${this.circleSize}px`,
            height: `${this.circleSize}px`
        };
    }
    /**
     * @return {?}
     */
    get circleStyle() {
        return {
            borderColor: this.color
        };
    }
    /**
     * @return {?}
     */
    get circleStyles() {
        /** @type {?} */
        const circlesPositions = [
            { top: 0, left: 0 },
            { left: `${this.circleSize * -0.36}px`, top: `${this.circleSize * 0.2}px` },
            { left: `${this.circleSize * -0.36}px`, top: `${this.circleSize * -0.2}px` },
            { left: 0, top: `${this.circleSize * -0.36}px` },
            { left: `${this.circleSize * 0.36}px`, top: `${this.circleSize * -0.2}px` },
            { left: `${this.circleSize * 0.36}px`, top: `${this.circleSize * 0.2}px` },
            { left: 0, top: `${this.circleSize * 0.36}px` }
        ];
        return circlesPositions.map((cp) => {
            return Object.assign({ cp }, this.circleStyle);
        });
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
IntersectingCirclesSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-intersecting-circles-spinner',
                template: `
    <div class="intersecting-circles-spinner" [ngStyle]="spinnerStyle">
      <div class="spinnerBlock" [ngStyle]="spinnerBlockStyle">
        <span class="circle" *ngFor="let cs of circleStyles" [ngStyle]="cs"></span>
      </div>
    </div>
  `,
                styles: [".intersecting-circles-spinner,.intersecting-circles-spinner *{box-sizing:border-box}.intersecting-circles-spinner{height:70px;width:70px;position:relative;display:flex;flex-direction:row;justify-content:center;align-items:center}.intersecting-circles-spinner .spinnerBlock{-webkit-animation:1.2s linear infinite intersecting-circles-spinners-animation;animation:1.2s linear infinite intersecting-circles-spinners-animation;-webkit-transform-origin:center;transform-origin:center;display:block;height:35px;width:35px}.intersecting-circles-spinner .circle{display:block;border:2px solid #ff1d5e;border-radius:50%;height:100%;width:100%;position:absolute;left:0;top:0}.intersecting-circles-spinner .circle:nth-child(1){left:0;top:0}.intersecting-circles-spinner .circle:nth-child(2){left:calc(35px * -.36);top:calc(35px * .2)}.intersecting-circles-spinner .circle:nth-child(3){left:calc(35px * -.36);top:calc(35px * -.2)}.intersecting-circles-spinner .circle:nth-child(4){left:0;top:calc(35px * -.36)}.intersecting-circles-spinner .circle:nth-child(5){left:calc(35px * .36);top:calc(35px * -.2)}.intersecting-circles-spinner .circle:nth-child(6){left:calc(35px * .36);top:calc(35px * .2)}.intersecting-circles-spinner .circle:nth-child(7){left:0;top:calc(35px * .36)}@-webkit-keyframes intersecting-circles-spinners-animation{from{-webkit-transform:rotate(0);transform:rotate(0)}to{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes intersecting-circles-spinners-animation{from{-webkit-transform:rotate(0);transform:rotate(0)}to{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
            }] }
];
/** @nocollapse */
IntersectingCirclesSpinnerComponent.ctorParameters = () => [];
IntersectingCirclesSpinnerComponent.propDecorators = {
    animationDuration: [{ type: Input }],
    size: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    IntersectingCirclesSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    IntersectingCirclesSpinnerComponent.prototype.size;
    /** @type {?} */
    IntersectingCirclesSpinnerComponent.prototype.color;
}
