// angular-epic-spinners/esm2015/src/app/spring-spinner/spring-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class SpringSpinnerComponent {
    constructor() {
        this.size = 60;
        this.color = '#fff';
        this.animationDuration = 2000;
        this.animationBaseName = 'spring-spinner-animation';
        this.currentAnimationName = '';
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`
        };
    }
    /**
     * @return {?}
     */
    get spinnerPartStyle() {
        return {
            height: `${this.size / 2}px`,
            width: `${this.size}px`
        };
    }
    /**
     * @return {?}
     */
    get rotatorStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`,
            borderRightColor: this.color,
            borderTopColor: this.color,
            borderWidth: `${this.size / 7}px`,
            animationDuration: `${this.animationDuration}ms`,
            animationName: this.currentAnimationName
        };
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
SpringSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-spring-spinner',
                template: `
    <div class="spring-spinner" [ngStyle]="spinnerStyle">
      <div class="spring-spinner-part top" [ngStyle]="spinnerPartStyle">
        <div class="spring-spinner-rotator" [ngStyle]="rotatorStyle"></div>
      </div>
      <div class="spring-spinner-part bottom" [ngStyle]="spinnerPartStyle">
        <div class="spring-spinner-rotator" [ngStyle]="rotatorStyle"></div>
      </div>
    </div>
  `,
                styles: [".spring-spinner,.spring-spinner *{box-sizing:border-box}.spring-spinner{height:60px;width:60px}.spring-spinner .spring-spinner-part{overflow:hidden;height:calc(60px / 2);width:60px}.spring-spinner .spring-spinner-part.bottom{-webkit-transform:rotate(180deg) scale(-1,1);transform:rotate(180deg) scale(-1,1)}.spring-spinner .spring-spinner-rotator{width:60px;height:60px;border:solid transparent;border-right-color:#ff1d5e;border-top-color:#ff1d5e;border-radius:50%;box-sizing:border-box;-webkit-animation:3s ease-in-out infinite spring-spinner-animation;animation:3s ease-in-out infinite spring-spinner-animation;-webkit-transform:rotate(-200deg);transform:rotate(-200deg)}@-webkit-keyframes spring-spinner-animation{0%,100%{border-width:calc(60px / 7)}25%,75%{border-width:calc(60px / 23.33)}50%{-webkit-transform:rotate(115deg);transform:rotate(115deg);border-width:calc(60px / 7)}}@keyframes spring-spinner-animation{0%,100%{border-width:calc(60px / 7)}25%,75%{border-width:calc(60px / 23.33)}50%{-webkit-transform:rotate(115deg);transform:rotate(115deg);border-width:calc(60px / 7)}}"]
            }] }
];
/** @nocollapse */
SpringSpinnerComponent.ctorParameters = () => [];
SpringSpinnerComponent.propDecorators = {
    size: [{ type: Input }],
    color: [{ type: Input }],
    animationDuration: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    SpringSpinnerComponent.prototype.size;
    /** @type {?} */
    SpringSpinnerComponent.prototype.color;
    /** @type {?} */
    SpringSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    SpringSpinnerComponent.prototype.animationBaseName;
    /** @type {?} */
    SpringSpinnerComponent.prototype.currentAnimationName;
}
