// angular-epic-spinners/esm2015/src/app/breeding-rhombus-spinner/breeding-rhombus-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class BreedingRhombusSpinnerComponent {
    constructor() {
        this.size = 150;
        this.animationDuration = 2000;
        this.color = '#fff';
        this.animationBaseName = 'breeding-rhombus-spinner-animation-child';
        this.rhombusesNum = 8;
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`
        };
    }
    /**
     * @return {?}
     */
    get rhombusStyle() {
        return {
            height: `${this.size / 7.5}px`,
            width: `${this.size / 7.5}px`,
            animationDuration: `${this.animationDuration}`,
            top: `${this.size / 2.3077}px`,
            left: `${this.size / 23077}px`,
            backgroundColor: this.color
        };
    }
    /**
     * @return {?}
     */
    get rhombusesStyles() {
        /** @type {?} */
        const rhombusesStyles = [];
        /** @type {?} */
        const delayModifier = this.animationDuration * 0.05;
        for (let i = 1; i <= this.rhombusesNum; i++) {
            rhombusesStyles.push(Object.assign({}, this.rhombusStyle, { animationDelay: `${delayModifier * (i + 1)}ms` }));
        }
        return rhombusesStyles;
    }
    /**
     * @return {?}
     */
    get bigRhombusStyle() {
        return {
            height: `${this.size / 3}px`,
            width: `${this.size / 3}px`,
            animationDuration: `${this.animationDuration}`,
            top: `${this.size / 3}px`,
            left: `${this.size / 3}px`,
            backgroundColor: this.color
        };
    }
    /**
     * @param {?} index
     * @return {?}
     */
    childClassString(index) {
        return `child-${index + 1}`;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
BreedingRhombusSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-breeding-rhombus-spinner',
                template: `
      <div class="breeding-rhombus-spinner" [ngStyle]="spinnerStyle">
        <div class="rhombus" *ngFor="let rs of rhombusesStyles; let i = index;" [ngStyle]="rs" [ngClass]="childClassString(i)"></div>
        <div class="rhombus big" [ngStyle]="bigRhombusStyle"></div>
      </div>
    `,
                styles: [".breeding-rhombus-spinner{height:65px;width:65px;position:relative;-webkit-transform:rotate(45deg);transform:rotate(45deg)}.breeding-rhombus-spinner,.breeding-rhombus-spinner *{box-sizing:border-box}.breeding-rhombus-spinner .rhombus{height:calc(65px / 7.5);width:calc(65px / 7.5);-webkit-animation-duration:2s;animation-duration:2s;top:calc(65px / 2.3077);left:calc(65px / 2.3077);background-color:#ff1d5e;position:absolute;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite}.breeding-rhombus-spinner .rhombus:nth-child(2n+0){margin-right:0}.breeding-rhombus-spinner .rhombus.child-1{-webkit-animation-name:breeding-rhombus-spinner-animation-child-1;animation-name:breeding-rhombus-spinner-animation-child-1;-webkit-animation-delay:calc(100ms * 1);animation-delay:calc(100ms * 1)}.breeding-rhombus-spinner .rhombus.child-2{-webkit-animation-name:breeding-rhombus-spinner-animation-child-2;animation-name:breeding-rhombus-spinner-animation-child-2;-webkit-animation-delay:calc(100ms * 2);animation-delay:calc(100ms * 2)}.breeding-rhombus-spinner .rhombus.child-3{-webkit-animation-name:breeding-rhombus-spinner-animation-child-3;animation-name:breeding-rhombus-spinner-animation-child-3;-webkit-animation-delay:calc(100ms * 3);animation-delay:calc(100ms * 3)}.breeding-rhombus-spinner .rhombus.child-4{-webkit-animation-name:breeding-rhombus-spinner-animation-child-4;animation-name:breeding-rhombus-spinner-animation-child-4;-webkit-animation-delay:calc(100ms * 4);animation-delay:calc(100ms * 4)}.breeding-rhombus-spinner .rhombus.child-5{-webkit-animation-name:breeding-rhombus-spinner-animation-child-5;animation-name:breeding-rhombus-spinner-animation-child-5;-webkit-animation-delay:calc(100ms * 5);animation-delay:calc(100ms * 5)}.breeding-rhombus-spinner .rhombus.child-6{-webkit-animation-name:breeding-rhombus-spinner-animation-child-6;animation-name:breeding-rhombus-spinner-animation-child-6;-webkit-animation-delay:calc(100ms * 6);animation-delay:calc(100ms * 6)}.breeding-rhombus-spinner .rhombus.child-7{-webkit-animation-name:breeding-rhombus-spinner-animation-child-7;animation-name:breeding-rhombus-spinner-animation-child-7;-webkit-animation-delay:calc(100ms * 7);animation-delay:calc(100ms * 7)}.breeding-rhombus-spinner .rhombus.child-8{-webkit-animation-name:breeding-rhombus-spinner-animation-child-8;animation-name:breeding-rhombus-spinner-animation-child-8;-webkit-animation-delay:calc(100ms * 8);animation-delay:calc(100ms * 8)}.breeding-rhombus-spinner .rhombus.big{height:calc(65px / 3);width:calc(65px / 3);top:calc(65px / 3);left:calc(65px / 3);background-color:#ff1d5e;-webkit-animation:2s .5s infinite breeding-rhombus-spinner-animation-child-big;animation:2s .5s infinite breeding-rhombus-spinner-animation-child-big}@-webkit-keyframes breeding-rhombus-spinner-animation-child-1{50%{-webkit-transform:translate(-325%,-325%);transform:translate(-325%,-325%)}}@keyframes breeding-rhombus-spinner-animation-child-1{50%{-webkit-transform:translate(-325%,-325%);transform:translate(-325%,-325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-2{50%{-webkit-transform:translate(0,-325%);transform:translate(0,-325%)}}@keyframes breeding-rhombus-spinner-animation-child-2{50%{-webkit-transform:translate(0,-325%);transform:translate(0,-325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-3{50%{-webkit-transform:translate(325%,-325%);transform:translate(325%,-325%)}}@keyframes breeding-rhombus-spinner-animation-child-3{50%{-webkit-transform:translate(325%,-325%);transform:translate(325%,-325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-4{50%{-webkit-transform:translate(325%,0);transform:translate(325%,0)}}@keyframes breeding-rhombus-spinner-animation-child-4{50%{-webkit-transform:translate(325%,0);transform:translate(325%,0)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-5{50%{-webkit-transform:translate(325%,325%);transform:translate(325%,325%)}}@keyframes breeding-rhombus-spinner-animation-child-5{50%{-webkit-transform:translate(325%,325%);transform:translate(325%,325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-6{50%{-webkit-transform:translate(0,325%);transform:translate(0,325%)}}@keyframes breeding-rhombus-spinner-animation-child-6{50%{-webkit-transform:translate(0,325%);transform:translate(0,325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-7{50%{-webkit-transform:translate(-325%,325%);transform:translate(-325%,325%)}}@keyframes breeding-rhombus-spinner-animation-child-7{50%{-webkit-transform:translate(-325%,325%);transform:translate(-325%,325%)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-8{50%{-webkit-transform:translate(-325%,0);transform:translate(-325%,0)}}@keyframes breeding-rhombus-spinner-animation-child-8{50%{-webkit-transform:translate(-325%,0);transform:translate(-325%,0)}}@-webkit-keyframes breeding-rhombus-spinner-animation-child-big{50%{-webkit-transform:scale(.5);transform:scale(.5)}}@keyframes breeding-rhombus-spinner-animation-child-big{50%{-webkit-transform:scale(.5);transform:scale(.5)}}"]
            }] }
];
/** @nocollapse */
BreedingRhombusSpinnerComponent.ctorParameters = () => [];
BreedingRhombusSpinnerComponent.propDecorators = {
    size: [{ type: Input }],
    animationDuration: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    BreedingRhombusSpinnerComponent.prototype.size;
    /** @type {?} */
    BreedingRhombusSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    BreedingRhombusSpinnerComponent.prototype.color;
    /** @type {?} */
    BreedingRhombusSpinnerComponent.prototype.animationBaseName;
    /** @type {?} */
    BreedingRhombusSpinnerComponent.prototype.rhombusesNum;
}
