// angular-epic-spinners/esm5/src/app/pixel-spinner/pixel-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var PixelSpinnerComponent = /** @class */ (function () {
    function PixelSpinnerComponent() {
        this.animationDuration = 1000;
        this.size = 50;
        this.color = '#fff';
        this.animationBaseName = 'pixel-spinner-animation';
        this.currentAnimationName = '';
    }
    Object.defineProperty(PixelSpinnerComponent.prototype, "pixelSize", {
        get: /**
         * @return {?}
         */
        function () {
            return this.size / 7;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(PixelSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                width: this.size + "px",
                height: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(PixelSpinnerComponent.prototype, "spinnerInnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                animationDuration: this.animationDuration + "ms",
                animationName: this.currentAnimationName,
                width: this.pixelSize + "px",
                height: this.pixelSize + "px",
                backgroundColor: this.color,
                color: this.color,
                boxShadow: "\n                        " + this.pixelSize * 1.5 + "px " + this.pixelSize * 1.5 + "px 0 0,\n                        " + this.pixelSize * -1.5 + "px " + this.pixelSize * -1.5 + "px 0 0,\n                        " + this.pixelSize * 1.5 + "px " + this.pixelSize * -1.5 + "px 0 0,\n                        " + this.pixelSize * -1.5 + "px " + this.pixelSize * 1.5 + "px 0 0,\n                        0 " + this.pixelSize * 1.5 + "px 0 0,\n                        " + this.pixelSize * 1.5 + "px 0 0 0,\n                        " + this.pixelSize * -1.5 + "px 0 0 0,\n                        0 " + this.pixelSize * -1.5 + "px 0 0\n                      "
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    PixelSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    PixelSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-pixel-spinner',
                    template: "\n    <div class=\"pixel-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"pixel-spinner-inner\" [ngStyle]=\"spinnerInnerStyle\"></div>\n    </div>\n  ",
                    styles: [".pixel-spinner,.pixel-spinner *{box-sizing:border-box}.pixel-spinner{height:70px;width:70px;display:flex;flex-direction:row;justify-content:center;align-items:center}.pixel-spinner .pixel-spinner-inner{width:calc(70px / 7);height:calc(70px / 7);background-color:#ff1d5e;color:#ff1d5e;box-shadow:15px 15px 0 0,-15px -15px 0 0,15px -15px 0 0,-15px 15px 0 0,0 15px 0 0,15px 0 0 0,-15px 0 0 0,0 -15px 0 0;-webkit-animation:2s linear infinite pixel-spinner-animation;animation:2s linear infinite pixel-spinner-animation}@-webkit-keyframes pixel-spinner-animation{50%,75%{box-shadow:20px 20px 0 0,-20px -20px 0 0,20px -20px 0 0,-20px 20px 0 0,0 10px 0 0,10px 0 0 0,-10px 0 0 0,0 -10px 0 0}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes pixel-spinner-animation{50%,75%{box-shadow:20px 20px 0 0,-20px -20px 0 0,20px -20px 0 0,-20px 20px 0 0,0 10px 0 0,10px 0 0 0,-10px 0 0 0,0 -10px 0 0}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
                }] }
    ];
    /** @nocollapse */
    PixelSpinnerComponent.ctorParameters = function () { return []; };
    PixelSpinnerComponent.propDecorators = {
        animationDuration: [{ type: Input }],
        size: [{ type: Input }],
        color: [{ type: Input }]
    };
    return PixelSpinnerComponent;
}());
export { PixelSpinnerComponent };
if (false) {
    /** @type {?} */
    PixelSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    PixelSpinnerComponent.prototype.size;
    /** @type {?} */
    PixelSpinnerComponent.prototype.color;
    /** @type {?} */
    PixelSpinnerComponent.prototype.animationBaseName;
    /** @type {?} */
    PixelSpinnerComponent.prototype.currentAnimationName;
}
