// angular-epic-spinners/esm2015/src/app/atom-spinner/atom-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class AtomSpinnerComponent {
    constructor() {
        this.size = 60;
        this.animationDuration = 1000;
        this.color = 'red';
    }
    /**
     * @return {?}
     */
    get spinnerStyle() {
        return {
            height: `${this.size}px`,
            width: `${this.size}px`
        };
    }
    /**
     * @return {?}
     */
    get circleStyle() {
        return {
            color: this.color,
            fontSize: `${this.size * 0.24}px`
        };
    }
    /**
     * @return {?}
     */
    get lineStyle() {
        return {
            animationDuration: `${this.animationDuration}ms`,
            borderLeftWidth: `${this.size / 25}px`,
            borderRightWidth: `${this.size / 25}px`,
            borderLeftColor: this.color
        };
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
AtomSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-atom-spinner',
                template: `
      <div class="atom-spinner" [ngStyle]="spinnerStyle">
        <div class="spinner-inner">
          <div class="spinner-line" [ngStyle]="lineStyle"></div>
          <div class="spinner-line" [ngStyle]="lineStyle"></div>
          <div class="spinner-line" [ngStyle]="lineStyle"></div>
          <div class="spinner-circle" [ngStyle]="circleStyle">
            &#9679;
          </div>
        </div>
      </div>
  `,
                styles: [".atom-spinner,.atom-spinner *{box-sizing:border-box}.atom-spinner{height:60px;width:60px;overflow:hidden}.atom-spinner .spinner-inner{position:relative;display:block;height:100%;width:100%}.atom-spinner .spinner-circle{display:block;position:absolute;color:#ff1d5e;font-size:calc(60px * .24);top:50%;left:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%)}.atom-spinner .spinner-line{position:absolute;width:100%;height:100%;border-radius:50%;-webkit-animation-duration:1s;animation-duration:1s;border-left-width:calc(60px / 25);border-top-width:calc(60px / 25);border-left-color:#ff1d5e;border-left-style:solid;border-top-style:solid;border-top-color:transparent}.atom-spinner .spinner-line:nth-child(1){-webkit-animation:1s linear infinite atom-spinner-animation-1;animation:1s linear infinite atom-spinner-animation-1;-webkit-transform:rotateZ(120deg) rotateX(66deg) rotateZ(0);transform:rotateZ(120deg) rotateX(66deg) rotateZ(0)}.atom-spinner .spinner-line:nth-child(2){-webkit-animation:1s linear infinite atom-spinner-animation-2;animation:1s linear infinite atom-spinner-animation-2;-webkit-transform:rotateZ(240deg) rotateX(66deg) rotateZ(0);transform:rotateZ(240deg) rotateX(66deg) rotateZ(0)}.atom-spinner .spinner-line:nth-child(3){-webkit-animation:1s linear infinite atom-spinner-animation-3;animation:1s linear infinite atom-spinner-animation-3;-webkit-transform:rotateZ(360deg) rotateX(66deg) rotateZ(0);transform:rotateZ(360deg) rotateX(66deg) rotateZ(0)}@-webkit-keyframes atom-spinner-animation-1{100%{-webkit-transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg)}}@keyframes atom-spinner-animation-1{100%{-webkit-transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(120deg) rotateX(66deg) rotateZ(360deg)}}@-webkit-keyframes atom-spinner-animation-2{100%{-webkit-transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg)}}@keyframes atom-spinner-animation-2{100%{-webkit-transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(240deg) rotateX(66deg) rotateZ(360deg)}}@-webkit-keyframes atom-spinner-animation-3{100%{-webkit-transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg)}}@keyframes atom-spinner-animation-3{100%{-webkit-transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg);transform:rotateZ(360deg) rotateX(66deg) rotateZ(360deg)}}"]
            }] }
];
/** @nocollapse */
AtomSpinnerComponent.ctorParameters = () => [];
AtomSpinnerComponent.propDecorators = {
    size: [{ type: Input }],
    animationDuration: [{ type: Input }],
    color: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    AtomSpinnerComponent.prototype.size;
    /** @type {?} */
    AtomSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    AtomSpinnerComponent.prototype.color;
}
