// angular-epic-spinners/esm5/src/app/flower-spinner/flower-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var FlowerSpinnerComponent = /** @class */ (function () {
    function FlowerSpinnerComponent() {
        this.size = 70;
        this.animationDuration = 2500;
        this.color = '#fff';
        this.smallerDotAnimationBaseName = 'flower-spinner-smaller-dot-animation';
        this.biggerDotAnimationBaseName = 'flower-spinner-bigger-dot-animation';
        this.currentSmallerDotAnimationBaseName = '';
        this.currentBiggerDotAnimationBaseName = '';
    }
    Object.defineProperty(FlowerSpinnerComponent.prototype, "dotSize", {
        get: /**
         * @return {?}
         */
        function () {
            return this.size / 7;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlowerSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlowerSpinnerComponent.prototype, "dotsContainerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                width: this.color + "px",
                height: this.color + "px",
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlowerSpinnerComponent.prototype, "smallerDotStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                background: this.color,
                animationDuration: this.animationDuration + "ms",
                animationName: this.currentSmallerDotAnimationBaseName
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlowerSpinnerComponent.prototype, "biggerDotStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                background: this.color,
                animationDuration: this.animationDuration + "ms",
                animationName: this.currentBiggerDotAnimationBaseName
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    FlowerSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    FlowerSpinnerComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
    };
    FlowerSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-flower-spinner',
                    template: "\n    <div class=\"flower-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"dots-container\" [ngStyle]=\"dotsContainerStyle\">\n        <div class=\"bigger-dot\" [ngStyle]=\"biggerDotStyle\">\n          <div class=\"smaller-dot\" [ngStyle]=\"smallerDotStyle\"></div>\n        </div>\n      </div>\n    </div>\n  ",
                    styles: [".flower-spinner,.flower-spinner *{box-sizing:border-box}.flower-spinner{height:70px;width:70px;display:flex;flex-direction:row;align-items:center;justify-content:center}.flower-spinner .dots-container{height:calc(70px / 7);width:calc(70px / 7)}.flower-spinner .smaller-dot{background:#ff1d5e;height:100%;width:100%;border-radius:50%;-webkit-animation:2.5s infinite both flower-spinner-smaller-dot-animation;animation:2.5s infinite both flower-spinner-smaller-dot-animation}.flower-spinner .bigger-dot{background:#ff1d5e;height:100%;width:100%;padding:10%;border-radius:50%;-webkit-animation:2.5s infinite both flower-spinner-bigger-dot-animation;animation:2.5s infinite both flower-spinner-bigger-dot-animation}@-webkit-keyframes flower-spinner-bigger-dot-animation{0%,100%{box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}25%,75%{box-shadow:#ff1d5e 26px 0 0,#ff1d5e -26px 0 0,#ff1d5e 0 26px 0,#ff1d5e 0 -26px 0,#ff1d5e 19px -19px 0,#ff1d5e 19px 19px 0,#ff1d5e -19px -19px 0,#ff1d5e -19px 19px 0}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg);box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}}@keyframes flower-spinner-bigger-dot-animation{0%,100%{box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}50%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}25%,75%{box-shadow:#ff1d5e 26px 0 0,#ff1d5e -26px 0 0,#ff1d5e 0 26px 0,#ff1d5e 0 -26px 0,#ff1d5e 19px -19px 0,#ff1d5e 19px 19px 0,#ff1d5e -19px -19px 0,#ff1d5e -19px 19px 0}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg);box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}}@-webkit-keyframes flower-spinner-smaller-dot-animation{0%,100%{box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}25%,75%{box-shadow:#ff1d5e 14px 0 0,#ff1d5e -14px 0 0,#ff1d5e 0 14px 0,#ff1d5e 0 -14px 0,#ff1d5e 10px -10px 0,#ff1d5e 10px 10px 0,#ff1d5e -10px -10px 0,#ff1d5e -10px 10px 0}}@keyframes flower-spinner-smaller-dot-animation{0%,100%{box-shadow:#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0,#ff1d5e 0 0 0}25%,75%{box-shadow:#ff1d5e 14px 0 0,#ff1d5e -14px 0 0,#ff1d5e 0 14px 0,#ff1d5e 0 -14px 0,#ff1d5e 10px -10px 0,#ff1d5e 10px 10px 0,#ff1d5e -10px -10px 0,#ff1d5e -10px 10px 0}}"]
                }] }
    ];
    /** @nocollapse */
    FlowerSpinnerComponent.ctorParameters = function () { return []; };
    FlowerSpinnerComponent.propDecorators = {
        size: [{ type: Input }],
        animationDuration: [{ type: Input }],
        color: [{ type: Input }]
    };
    return FlowerSpinnerComponent;
}());
export { FlowerSpinnerComponent };
if (false) {
    /** @type {?} */
    FlowerSpinnerComponent.prototype.size;
    /** @type {?} */
    FlowerSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    FlowerSpinnerComponent.prototype.color;
    /** @type {?} */
    FlowerSpinnerComponent.prototype.smallerDotAnimationBaseName;
    /** @type {?} */
    FlowerSpinnerComponent.prototype.biggerDotAnimationBaseName;
    /** @type {?} */
    FlowerSpinnerComponent.prototype.currentSmallerDotAnimationBaseName;
    /** @type {?} */
    FlowerSpinnerComponent.prototype.currentBiggerDotAnimationBaseName;
}
