// angular-epic-spinners/esm5/src/app/scaling-squares-spinner/scaling-squares-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
var ScalingSquaresSpinnerComponent = /** @class */ (function () {
    function ScalingSquaresSpinnerComponent() {
        this.size = 65;
        this.color = '#fff';
        this.animationDuration = 1250;
        this.squaresNum = 4;
    }
    Object.defineProperty(ScalingSquaresSpinnerComponent.prototype, "spinnerStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScalingSquaresSpinnerComponent.prototype, "squareStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                height: this.size * 0.25 / 1.3 + "px",
                width: this.size * 0.25 / 1.3 + "px",
                animationDuration: this.animationDuration + "ms",
                borderWidth: this.size * 0.04 / 1.3 + "px",
                borderColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScalingSquaresSpinnerComponent.prototype, "squaresStyles", {
        get: /**
         * @return {?}
         */
        function () {
            /** @type {?} */
            var squaresStyles = [];
            for (var i = 1; i <= this.squaresNum; i++) {
                squaresStyles.push(tslib_1.__assign({}, this.squareStyle));
            }
            return squaresStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} i
     * @return {?}
     */
    ScalingSquaresSpinnerComponent.prototype.childClass = /**
     * @param {?} i
     * @return {?}
     */
    function (i) {
        return "square-" + (i + 1);
    };
    /**
     * @return {?}
     */
    ScalingSquaresSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    ScalingSquaresSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-scaling-squares-spinner',
                    template: "\n    <div class=\"scaling-squares-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"square\"\n           *ngFor=\"let ss of squaresStyles; let i = index;\"\n           [ngClass]=\"childClass(i)\"\n           [ngStyle]=\"ss\">\n      </div>\n    </div>\n  ",
                    styles: [".scaling-squares-spinner,.scaling-squares-spinner *{box-sizing:border-box}.scaling-squares-spinner{height:65px;width:65px;position:relative;display:flex;flex-direction:row;align-items:center;justify-content:center;-webkit-animation:1.25s infinite scaling-squares-animation;animation:1.25s infinite scaling-squares-animation;-webkit-transform:rotate(0);transform:rotate(0)}.scaling-squares-spinner .square{height:calc(65px * .25 / 1.3);width:calc(65px * .25 / 1.3);margin-right:auto;margin-left:auto;border:solid #ff1d5e;position:absolute;-webkit-animation-duration:1.25s;animation-duration:1.25s;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite}.scaling-squares-spinner .square:nth-child(1){-webkit-animation-name:scaling-squares-spinner-animation-child-1;animation-name:scaling-squares-spinner-animation-child-1}.scaling-squares-spinner .square:nth-child(2){-webkit-animation-name:scaling-squares-spinner-animation-child-2;animation-name:scaling-squares-spinner-animation-child-2}.scaling-squares-spinner .square:nth-child(3){-webkit-animation-name:scaling-squares-spinner-animation-child-3;animation-name:scaling-squares-spinner-animation-child-3}.scaling-squares-spinner .square:nth-child(4){-webkit-animation-name:scaling-squares-spinner-animation-child-4;animation-name:scaling-squares-spinner-animation-child-4}@-webkit-keyframes scaling-squares-animation{50%{-webkit-transform:rotate(90deg);transform:rotate(90deg)}100%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}}@keyframes scaling-squares-animation{50%{-webkit-transform:rotate(90deg);transform:rotate(90deg)}100%{-webkit-transform:rotate(180deg);transform:rotate(180deg)}}@-webkit-keyframes scaling-squares-spinner-animation-child-1{50%{-webkit-transform:translate(150%,150%) scale(2,2);transform:translate(150%,150%) scale(2,2)}}@keyframes scaling-squares-spinner-animation-child-1{50%{-webkit-transform:translate(150%,150%) scale(2,2);transform:translate(150%,150%) scale(2,2)}}@-webkit-keyframes scaling-squares-spinner-animation-child-2{50%{-webkit-transform:translate(-150%,150%) scale(2,2);transform:translate(-150%,150%) scale(2,2)}}@keyframes scaling-squares-spinner-animation-child-2{50%{-webkit-transform:translate(-150%,150%) scale(2,2);transform:translate(-150%,150%) scale(2,2)}}@-webkit-keyframes scaling-squares-spinner-animation-child-3{50%{-webkit-transform:translate(-150%,-150%) scale(2,2);transform:translate(-150%,-150%) scale(2,2)}}@keyframes scaling-squares-spinner-animation-child-3{50%{-webkit-transform:translate(-150%,-150%) scale(2,2);transform:translate(-150%,-150%) scale(2,2)}}@-webkit-keyframes scaling-squares-spinner-animation-child-4{50%{-webkit-transform:translate(150%,-150%) scale(2,2);transform:translate(150%,-150%) scale(2,2)}}@keyframes scaling-squares-spinner-animation-child-4{50%{-webkit-transform:translate(150%,-150%) scale(2,2);transform:translate(150%,-150%) scale(2,2)}}"]
                }] }
    ];
    /** @nocollapse */
    ScalingSquaresSpinnerComponent.ctorParameters = function () { return []; };
    ScalingSquaresSpinnerComponent.propDecorators = {
        size: [{ type: Input }],
        color: [{ type: Input }],
        animationDuration: [{ type: Input }]
    };
    return ScalingSquaresSpinnerComponent;
}());
export { ScalingSquaresSpinnerComponent };
if (false) {
    /** @type {?} */
    ScalingSquaresSpinnerComponent.prototype.size;
    /** @type {?} */
    ScalingSquaresSpinnerComponent.prototype.color;
    /** @type {?} */
    ScalingSquaresSpinnerComponent.prototype.animationDuration;
    /** @type {?} */
    ScalingSquaresSpinnerComponent.prototype.squaresNum;
}
